// rtk-ui-kit/components/PickerDate/CalendarDates/styledGenerators.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.generateGradient =
  exports.generateColor =
  exports.generateBackground =
  exports.generateShape =
  exports.generateShapeActiveItem =
    void 0;

var styled_components_1 = require('styled-components');

var components_1 = require('../../../constants/components');

var constants_1 = require('./constants');

var getBorderRadius = function getBorderRadius(shape) {
  switch (shape) {
    case components_1.SHAPES.geometric:
      return 0;

    case components_1.SHAPES.rounded:
      return 8;

    case components_1.SHAPES.circular:
      return 36;

    default:
      return 0;
  }
};

var generateShapeActiveItem = function generateShapeActiveItem(_a) {
  var shape = _a.shape;
  return styled_components_1.css(
    templateObject_1 ||
      (templateObject_1 = __makeTemplateObject(
        ['\n    border-radius: ', 'px;\n  '],
        ['\n    border-radius: ', 'px;\n  ']
      )),
    getBorderRadius(shape)
  );
};

exports.generateShapeActiveItem = generateShapeActiveItem;

var generateShape = function generateShape(_a) {
  var shape = _a.shape,
    status = _a.status;
  var borderRadius = getBorderRadius(shape);

  if (status === constants_1.DATE_STATUS.focusedFirst) {
    return styled_components_1.css(
      templateObject_2 ||
        (templateObject_2 = __makeTemplateObject(
          [
            '\n      border-top-left-radius: ',
            'px;\n      border-bottom-left-radius: ',
            'px;\n    ',
          ],
          [
            '\n      border-top-left-radius: ',
            'px;\n      border-bottom-left-radius: ',
            'px;\n    ',
          ]
        )),
      borderRadius,
      borderRadius
    );
  }

  if (status === constants_1.DATE_STATUS.focusedLast) {
    return styled_components_1.css(
      templateObject_3 ||
        (templateObject_3 = __makeTemplateObject(
          [
            '\n      border-top-right-radius: ',
            'px;\n      border-bottom-right-radius: ',
            'px;\n    ',
          ],
          [
            '\n      border-top-right-radius: ',
            'px;\n      border-bottom-right-radius: ',
            'px;\n    ',
          ]
        )),
      borderRadius,
      borderRadius
    );
  }

  if (status === constants_1.DATE_STATUS.focused) {
    return styled_components_1.css(
      templateObject_4 ||
        (templateObject_4 = __makeTemplateObject(
          ['\n      border-radius: 0px;\n    '],
          ['\n      border-radius: 0px;\n    ']
        ))
    );
  }

  return styled_components_1.css(
    templateObject_5 ||
      (templateObject_5 = __makeTemplateObject(
        ['\n    :hover {\n      border-radius: ', 'px;\n    }\n  '],
        ['\n    :hover {\n      border-radius: ', 'px;\n    }\n  ']
      )),
    borderRadius
  );
};

exports.generateShape = generateShape;

var generateBackground = function generateBackground(_a) {
  var theme = _a.theme,
    status = _a.status;

  switch (status) {
    case constants_1.DATE_STATUS.disabled:
      return styled_components_1.css(
        templateObject_6 ||
          (templateObject_6 = __makeTemplateObject(
            ['\n        :hover {\n          background: inherit;\n        }\n      '],
            ['\n        :hover {\n          background: inherit;\n        }\n      ']
          ))
      );

    case constants_1.DATE_STATUS.focused:
    case constants_1.DATE_STATUS.focusedFirst:
    case constants_1.DATE_STATUS.focusedLast:
      return styled_components_1.css(
        templateObject_7 ||
          (templateObject_7 = __makeTemplateObject(
            ['\n        background: ', ';\n      '],
            ['\n        background: ', ';\n      ']
          )),
        theme.transparent[5]
      );

    default:
      return styled_components_1.css(
        templateObject_8 ||
          (templateObject_8 = __makeTemplateObject(
            [
              '\n        background: inherit;\n        :hover {\n          background: ',
              ';\n        }\n      ',
            ],
            [
              '\n        background: inherit;\n        :hover {\n          background: ',
              ';\n        }\n      ',
            ]
          )),
        theme.transparent[5]
      );
  }
};

exports.generateBackground = generateBackground;

var generateColor = function generateColor(_a) {
  var theme = _a.theme,
    status = _a.status,
    color = _a.color,
    isCurrent = _a.isCurrent;

  if (isCurrent) {
    return styled_components_1.css(
      templateObject_9 ||
        (templateObject_9 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      color
    );
  }

  switch (status) {
    case constants_1.DATE_STATUS.disabled:
      return styled_components_1.css(
        templateObject_10 ||
          (templateObject_10 = __makeTemplateObject(
            ['\n        color: ', ';\n      '],
            ['\n        color: ', ';\n      ']
          )),
        theme.transparent[25]
      );

    default:
      return styled_components_1.css(
        templateObject_11 ||
          (templateObject_11 = __makeTemplateObject(
            ['\n        color: ', ';\n      '],
            ['\n        color: ', ';\n      ']
          )),
        theme.transparent[75]
      );
  }
};

exports.generateColor = generateColor;

var generateGradient = function generateGradient(_a) {
  var theme = _a.theme,
    gradient = _a.gradient;

  switch (gradient) {
    case constants_1.RANGE_GRADIENT.before:
      return styled_components_1.css(
        templateObject_12 ||
          (templateObject_12 = __makeTemplateObject(
            ['\n        background: linear-gradient(to left, ', ', transparent);\n      '],
            ['\n        background: linear-gradient(to left, ', ', transparent);\n      ']
          )),
        theme.transparent[5]
      );

    case constants_1.RANGE_GRADIENT.after:
      return styled_components_1.css(
        templateObject_13 ||
          (templateObject_13 = __makeTemplateObject(
            ['\n        background: linear-gradient(to right, ', ', transparent);\n      '],
            ['\n        background: linear-gradient(to right, ', ', transparent);\n      ']
          )),
        theme.transparent[5]
      );

    default:
      return styled_components_1.css(
        templateObject_14 ||
          (templateObject_14 = __makeTemplateObject(
            ['\n        background: unset;\n      '],
            ['\n        background: unset;\n      ']
          ))
      );
  }
};

exports.generateGradient = generateGradient;
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12,
  templateObject_13,
  templateObject_14;
