// rtk-ui-kit/components/PickerDate/CalendarDates/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledCalendarDateEmpty =
  exports.StyledCalendarDateActive =
  exports.StyledCalendarDateInner =
  exports.StyledCalendarDatesItem =
  exports.StyledCalendarDatesDayWeek =
  exports.StyledCalendarDates =
    void 0;

var styled_components_1 = __importDefault(require('styled-components'));

var styledGenerators_1 = require('./styledGenerators');

var constants_1 = require('./constants');

exports.StyledCalendarDates = styled_components_1.default.div(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  display: grid;\n  grid-template-columns: repeat(7, 1fr);\n  padding: 0px 14px 12px 14px;\n',
      ],
      [
        '\n  display: grid;\n  grid-template-columns: repeat(7, 1fr);\n  padding: 0px 14px 12px 14px;\n',
      ]
    ))
);
exports.StyledCalendarDatesDayWeek = styled_components_1.default.div(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      [
        "\n  font-size: 12px;\n  line-height: 16px;\n  font-feature-settings: 'tnum' on, 'lnum' on;\n  text-align: center;\n  padding: 0px 6px 4px 6px;\n  color: ",
        ';\n',
      ],
      [
        "\n  font-size: 12px;\n  line-height: 16px;\n  font-feature-settings: 'tnum' on, 'lnum' on;\n  text-align: center;\n  padding: 0px 6px 4px 6px;\n  color: ",
        ';\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[25];
  }
);
exports.StyledCalendarDatesItem = styled_components_1.default.div(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      [
        '\n  width: 36px;\n  font-size: 14px;\n  line-height: 20px;\n  cursor: ',
        ';\n  ',
        ';\n  ',
        ';\n  ',
        ';\n',
      ],
      [
        '\n  width: 36px;\n  font-size: 14px;\n  line-height: 20px;\n  cursor: ',
        ';\n  ',
        ';\n  ',
        ';\n  ',
        ';\n',
      ]
    )),
  function (_a) {
    var status = _a.status;
    return status === constants_1.DATE_STATUS.disabled ? 'default' : 'pointer';
  },
  styledGenerators_1.generateShape,
  styledGenerators_1.generateBackground,
  styledGenerators_1.generateColor
);
exports.StyledCalendarDateInner = styled_components_1.default.div(
  templateObject_4 ||
    (templateObject_4 = __makeTemplateObject(
      [
        '\n  height: 36px;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n}\n',
      ],
      [
        '\n  height: 36px;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n}\n',
      ]
    ))
);
exports.StyledCalendarDateActive = styled_components_1.default(exports.StyledCalendarDateInner)(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      ['\n  ', ';\n  color: ', ';\n  background: ', ';\n  :hover {\n    background: ', ';\n  }\n'],
      ['\n  ', ';\n  color: ', ';\n  background: ', ';\n  :hover {\n    background: ', ';\n  }\n']
    )),
  styledGenerators_1.generateShapeActiveItem,
  function (_a) {
    var theme = _a.theme;
    return theme.greyScale.white;
  },
  function (_a) {
    var color = _a.color;
    return color;
  },
  function (_a) {
    var color = _a.color;
    return color;
  }
);
exports.StyledCalendarDateEmpty = styled_components_1.default.div(
  templateObject_6 || (templateObject_6 = __makeTemplateObject(['\n  ', '\n'], ['\n  ', '\n'])),
  styledGenerators_1.generateGradient
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6;
