// rtk-ui-kit/components/Button/FunctionButton/FunctionButton.js
'use strict';
var PropTypes = require('prop-types');

var __assign =
  (void 0 && (void 0).__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];

          for (var p in s) {
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
          }
        }

        return t;
      };

    return __assign.apply(this, arguments);
  };

var __rest =
  (void 0 && (void 0).__rest) ||
  function (s, e) {
    var t = {};

    for (var p in s) {
      if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    }

    if (s != null && typeof Object.getOwnPropertySymbols === 'function')
      for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
          t[p[i]] = s[p[i]];
      }
    return t;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importDefault(require('react'));

var useGetBaseColorFromProps_1 = require('../../../hooks/useGetBaseColorFromProps');

var components_1 = require('../../../constants/components');

var styles_1 = require('./styles');

var FunctionButton = function FunctionButton(props) {
  var _a = props.color,
    color = _a === void 0 ? components_1.BASE_COLORS.primary2 : _a,
    _b = props.view,
    view = _b === void 0 ? components_1.VIEWS.primary : _b,
    icon = props.icon,
    _c = props.iconPosition,
    iconPosition = _c === void 0 ? components_1.ICON_POSITIONS.left : _c,
    children = props.children,
    restProps = __rest(props, ['color', 'view', 'icon', 'iconPosition', 'children']);

  var renderText = function renderText() {
    if (iconPosition === components_1.ICON_POSITIONS.left) {
      return react_1.default.createElement(
        react_1.default.Fragment,
        null,
        icon,
        children &&
          react_1.default.createElement(
            styles_1.StyledText,
            {
              variant: components_1.TYPOGRAPHY_VARIANTS.accentS,
              tag: 'span',
              isStatic: true,
            },
            children
          )
      );
    }

    return react_1.default.createElement(
      react_1.default.Fragment,
      null,
      children &&
        react_1.default.createElement(
          styles_1.StyledText,
          {
            variant: components_1.TYPOGRAPHY_VARIANTS.accentS,
            className: iconPosition,
            tag: 'span',
            isStatic: true,
          },
          children
        ),
      icon
    );
  };

  var baseColorFromProps = useGetBaseColorFromProps_1.useGetBaseColorFromProps(color);
  var buttonText = renderText();
  return react_1.default.createElement(
    styles_1.StyledFunctionButton,
    __assign(
      {
        color: baseColorFromProps,
        view: view,
      },
      restProps
    ),
    buttonText
  );
};

FunctionButton.propTypes = {
  children: PropTypes.node,
  /**
   * Задает основной цвет для кнопки
   *  @default "primary2"
   */
  color: PropTypes.oneOf(['primary1', 'primary2', 'secondary1', 'secondary2']),
  /**
   * Иконка для кнопки
   */
  icon: PropTypes.node,
  /**
   * Задает расположение иконки относительно текста
   *  @default "left"
   */
  iconPosition: PropTypes.oneOf(['left', 'right']),
  /**
   * Задает внешний вид
   *  @default "primary"
   */
  view: PropTypes.oneOf(['primary', 'secondary', 'tertiary']),
};

FunctionButton.displayName = 'FunctionButton';
exports.default = FunctionButton;
