// rtk-ui-kit/components/Notifications/CustomToastNotification/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledToastNotificationIconInner =
  exports.StyledToastNotificationIcon =
  exports.StyledToastNotificationActionButtons =
  exports.StyledToastNotificationContent =
  exports.StyledToastNotificationInner =
  exports.StyledToastNotification =
    void 0;

var styled_components_1 = __importStar(require('styled-components'));

var styledGenerators_1 = require('../styledGenerators');

var basicToastOut = styled_components_1.css(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  &.toast-notification-exit {\n    opacity: 0;\n  }\n  &.toast-notification-exit-done {\n    opacity: 0;\n  }\n',
      ],
      [
        '\n  &.toast-notification-exit {\n    opacity: 0;\n  }\n  &.toast-notification-exit-done {\n    opacity: 0;\n  }\n',
      ]
    ))
);
exports.StyledToastNotification = styled_components_1.default.div(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      [
        '\n  box-shadow: ',
        ';\n  width: 100%;\n  ',
        ';\n  background: ',
        ';\n  transition: background ',
        ',\n    transform ',
        ',\n    box-shadow ',
        ',\n    opacity ',
        ',\n    height ',
        ',\n    margin ',
        ';\n  ',
        ';\n  ',
        ';\n  margin: 6px 0;\n  overflow: hidden;\n  height: 0;\n',
      ],
      [
        '\n  box-shadow: ',
        ';\n  width: 100%;\n  ',
        ';\n  background: ',
        ';\n  transition: background ',
        ',\n    transform ',
        ',\n    box-shadow ',
        ',\n    opacity ',
        ',\n    height ',
        ',\n    margin ',
        ';\n  ',
        ';\n  ',
        ';\n  margin: 6px 0;\n  overflow: hidden;\n  height: 0;\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.shadows.bottomL;
  },
  styledGenerators_1.generateNotificationShape,
  function (_a) {
    var theme = _a.theme;
    return theme.toastNotifications;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transitions.easeInOut;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transitions.cubicBezier;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transitions.easeInOut;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transitions.easeInOut;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transitions.easeInOut;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transitions.easeInOut;
  },
  styledGenerators_1.generatePosition,
  basicToastOut
);
exports.StyledToastNotificationInner = styled_components_1.default.div(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      [
        '\n  width: 100%;\n  display: flex;\n  flex-wrap: wrap;\n  padding: 16px;\n  align-items: center;\n  box-sizing: border-box;\n',
      ],
      [
        '\n  width: 100%;\n  display: flex;\n  flex-wrap: wrap;\n  padding: 16px;\n  align-items: center;\n  box-sizing: border-box;\n',
      ]
    ))
);
exports.StyledToastNotificationContent = styled_components_1.default.div(
  templateObject_4 ||
    (templateObject_4 = __makeTemplateObject(
      [
        '\n  flex: 1;\n  margin: 0 16px;\n  display: grid;\n  flex-direction: column;\n  gap: 4px;\n',
      ],
      [
        '\n  flex: 1;\n  margin: 0 16px;\n  display: grid;\n  flex-direction: column;\n  gap: 4px;\n',
      ]
    ))
);
exports.StyledToastNotificationActionButtons = styled_components_1.default.div(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      [
        '\n  display: flex;\n  padding-top: 12px;\n  width: 100%;\n  flex-direction: row-reverse;\n  justify-content: flex-start;\n  align-items: center;\n  border-top: 1px solid ',
        ';\n  margin-top: 12px;\n',
      ],
      [
        '\n  display: flex;\n  padding-top: 12px;\n  width: 100%;\n  flex-direction: row-reverse;\n  justify-content: flex-start;\n  align-items: center;\n  border-top: 1px solid ',
        ';\n  margin-top: 12px;\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[5];
  }
);
exports.StyledToastNotificationIcon = styled_components_1.default.div(
  templateObject_6 ||
    (templateObject_6 = __makeTemplateObject(
      ['\n  align-self: flex-start;\n'],
      ['\n  align-self: flex-start;\n']
    ))
);
exports.StyledToastNotificationIconInner = styled_components_1.default.div(
  templateObject_7 ||
    (templateObject_7 = __makeTemplateObject(
      [
        '\n  width: 36px;\n  height: 36px;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  ',
        ';\n  ',
        ';\n  transition: background ',
        ';\n  & svg > path {\n    fill: ',
        ';\n  }\n  img {\n    max-width: 36px;\n    ',
        ';\n  }\n',
      ],
      [
        '\n  width: 36px;\n  height: 36px;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  ',
        ';\n  ',
        ';\n  transition: background ',
        ';\n  & svg > path {\n    fill: ',
        ';\n  }\n  img {\n    max-width: 36px;\n    ',
        ';\n  }\n',
      ]
    )),
  styledGenerators_1.generateIconBackground,
  styledGenerators_1.generateNotificationShape,
  function (_a) {
    var theme = _a.theme;
    return theme.transitions.easeInOut;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.greyScale.white;
  },
  styledGenerators_1.generateNotificationShape
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7;
