// rtk-ui-kit/components/Breadcrumbs/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledDropdown =
  exports.StyledMoreWrapper =
  exports.StyledDivider =
  exports.StyledTypography =
  exports.StyledBreadcrumbItem =
  exports.StyledBreadcrumbs =
    void 0;

var styled_components_1 = __importStar(require('styled-components'));

var Typography_1 = __importDefault(require('../Typography/Typography'));

var DropdownMenu_1 = __importDefault(require('../DropdownMenu/DropdownMenu'));

var components_1 = require('../../constants/components');

var generateShape = function generateShape(_a) {
  var shape = _a.shape;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_1 ||
          (templateObject_1 = __makeTemplateObject(
            ['\n        border-radius: 0px;\n      '],
            ['\n        border-radius: 0px;\n      ']
          ))
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_2 ||
          (templateObject_2 = __makeTemplateObject(
            ['\n        border-radius: 4px;\n      '],
            ['\n        border-radius: 4px;\n      ']
          ))
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_3 ||
          (templateObject_3 = __makeTemplateObject(
            ['\n        border-radius: 8px;\n      '],
            ['\n        border-radius: 8px;\n      ']
          ))
      );

    default:
      return '';
  }
};

exports.StyledBreadcrumbs = styled_components_1.default.div(
  templateObject_4 || (templateObject_4 = __makeTemplateObject([''], ['']))
);
exports.StyledBreadcrumbItem = styled_components_1.default.span(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      ['\n  display: inline-flex;\n  align-items: center;\n'],
      ['\n  display: inline-flex;\n  align-items: center;\n']
    ))
);
exports.StyledTypography = styled_components_1.default(Typography_1.default)(
  templateObject_6 ||
    (templateObject_6 = __makeTemplateObject(
      [
        '\n  color: ',
        ';\n  white-space: nowrap;\n\n  &:last-child {\n    color: ',
        ';\n    a {\n      color: ',
        ';\n      pointer-events: none;\n    }\n  }\n\n  a {\n    text-decoration: none;\n    color: ',
        ';\n  }\n',
      ],
      [
        '\n  color: ',
        ';\n  white-space: nowrap;\n\n  &:last-child {\n    color: ',
        ';\n    a {\n      color: ',
        ';\n      pointer-events: none;\n    }\n  }\n\n  a {\n    text-decoration: none;\n    color: ',
        ';\n  }\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[100];
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[50];
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[50];
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[100];
  }
);
exports.StyledDivider = styled_components_1.default.div(
  templateObject_7 ||
    (templateObject_7 = __makeTemplateObject(
      [
        '\n  display: inline-flex;\n  width: 3px;\n  height: 3px;\n  background: ',
        ';\n  border-radius: 50%;\n  margin: 0 8px;\n',
      ],
      [
        '\n  display: inline-flex;\n  width: 3px;\n  height: 3px;\n  background: ',
        ';\n  border-radius: 50%;\n  margin: 0 8px;\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[100];
  }
);
exports.StyledMoreWrapper = styled_components_1.default.div(
  templateObject_8 ||
    (templateObject_8 = __makeTemplateObject(
      [
        '\n  display: inline-flex;\n  margin: 0 8px;\n  cursor: pointer;\n  background-color: ',
        ';\n  ',
        ';\n',
      ],
      [
        '\n  display: inline-flex;\n  margin: 0 8px;\n  cursor: pointer;\n  background-color: ',
        ';\n  ',
        ';\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme,
      isOpen = _a.isOpen;
    return isOpen ? theme.transparent[5] : 'inherit';
  },
  generateShape
);
exports.StyledDropdown = styled_components_1.default(DropdownMenu_1.default)(
  templateObject_9 ||
    (templateObject_9 = __makeTemplateObject(
      ['\n  display: inline-flex;\n'],
      ['\n  display: inline-flex;\n']
    ))
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9;
