// rtk-ui-kit/components/TabsClassic/TabsClassicGroup/TabsClassicGroup.js
'use strict';
var PropTypes = require('prop-types');

var __assign =
  (void 0 && (void 0).__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];

          for (var p in s) {
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
          }
        }

        return t;
      };

    return __assign.apply(this, arguments);
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importStar(require('react'));

var constants_1 = require('../constants');

var context_1 = __importDefault(require('../context'));

var useGetBaseColorFromProps_1 = require('../../../hooks/useGetBaseColorFromProps');

var function_1 = require('../../../utils/function');

var styles_1 = require('./styles');

var TabsClassicGroup = function TabsClassicGroup(props) {
  var _a = props.accentColor,
    accentColor = _a === void 0 ? constants_1.DEFAULT_TAB_COLOR : _a,
    _b = props.size,
    size = _b === void 0 ? constants_1.DEFAULT_TAB_SIZE : _b,
    _c = props.disabledGroup,
    disabledGroup = _c === void 0 ? false : _c,
    value = props.value,
    _d = props.classicBar,
    classicBar = _d === void 0 ? true : _d,
    _e = props.scrollable,
    scrollable = _e === void 0 ? false : _e,
    className = props.className,
    style = props.style,
    _f = props.onChange,
    onChange = _f === void 0 ? function_1.noop : _f,
    children = props.children;
  var ref = react_1.useRef();

  var _g = react_1.useState(value),
    activeIndex = _g[0],
    setActiveIndex = _g[1];

  var _h = react_1.useState(false),
    isScrolling = _h[0],
    setIsScrolling = _h[1];

  var _j = react_1.useState(0),
    startX = _j[0],
    setStartX = _j[1];

  var _k = react_1.useState(0),
    scrollX = _k[0],
    setScrollX = _k[1];

  var color = useGetBaseColorFromProps_1.useGetBaseColorFromProps(accentColor);

  var handleChange = function handleChange(index) {
    setActiveIndex(index);
    onChange(index);
  };

  var onMouseDown = function onMouseDown(event) {
    setIsScrolling(true);
    var x = event.pageX - ref.current.offsetLeft;
    setStartX(x);
    setScrollX(ref.current.scrollLeft);
  };

  var onMouseUp = function onMouseUp() {
    ref.current.classList.remove('grab');
    setIsScrolling(false);
  };

  var onMouseLeave = function onMouseLeave() {
    setIsScrolling(false);
  };

  var onMouseMove = function onMouseMove(event) {
    if (!isScrolling) return;
    event.preventDefault();
    ref.current.classList.add('grab');
    var x = event.pageX - ref.current.offsetLeft;
    var step = x - startX;
    ref.current.scrollLeft = scrollX - step;
  };

  return react_1.default.createElement(
    context_1.default,
    {
      value: {
        size: size,
        scrollable: scrollable,
        disabledGroup: disabledGroup,
        activeIndex: activeIndex,
        handleChange: handleChange,
        accentColor: color,
        tabsRef: ref,
      },
    },

    react_1.default.createElement(
      styles_1.StyledTabsClassicGroupWrapper,
      __assign(
        {
          ref: ref,
        },
        scrollable && {
          onMouseDown: onMouseDown,
          onMouseUp: onMouseUp,
          onMouseMove: onMouseMove,
          onMouseLeave: onMouseLeave,
        },
        {
          className: className,
          style: style,
          scrollable: scrollable,
          role: 'tablist',
        }
      ),
      react_1.default.createElement(
        styles_1.StyledTabsClassicGroup,
        {
          size: size,
          classicBar: classicBar,
          scrollable: scrollable,
        },
        children
      )
    )
  );
};

TabsClassicGroup.propTypes = {
  /**
   * Задает основной цвет для активной кнопки группы
   *  @default "primary2"
   */
  accentColor: PropTypes.oneOf(['primary1', 'primary2', 'secondary1', 'secondary2']),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.element),
    PropTypes.object,
    PropTypes.shape({
      /**
       * Iterator
       */
      '__@iterator': PropTypes.func.isRequired,
      /**
       * Returns an object whose properties have the value 'true'
       * when they will be absent when used in a 'with' statement.
       */
      '__@unscopables': PropTypes.func.isRequired,
      /**
       * Combines two or more arrays.
       * @param items Additional items to add to the end of array1.
       */
      concat: PropTypes.func.isRequired,
      /**
       * Returns the this object after copying a section of the array identified by start and end
       * to the same array starting at position target
       * @param target If target is negative, it is treated as length+target where length is the
       * length of the array.
       * @param start If start is negative, it is treated as length+start. If end is negative, it
       * is treated as length+end.
       * @param end If not specified, length of the this object is used as its default value.
       */
      copyWithin: PropTypes.func.isRequired,
      /**
       * Returns an iterable of key, value pairs for every entry in the array
       */
      entries: PropTypes.func.isRequired,
      /**
       * Determines whether all the members of an array satisfy the specified test.
       * @param callbackfn A function that accepts up to three arguments. The every method calls
       * the callbackfn function for each element in the array until the callbackfn returns a value
       * which is coercible to the Boolean value false, or until the end of the array.
       * @param thisArg An object to which the this keyword can refer in the callbackfn function.
       * If thisArg is omitted, undefined is used as the this value.
       */
      every: PropTypes.func.isRequired,
      /**
       * Returns the this object after filling the section identified by start and end with value
       * @param value value to fill array section with
       * @param start index to start filling the array at. If start is negative, it is treated as
       * length+start where length is the length of the array.
       * @param end index to stop filling the array at. If end is negative, it is treated as
       * length+end.
       */
      fill: PropTypes.func.isRequired,
      /**
       * Returns the elements of an array that meet the condition specified in a callback function.
       * @param callbackfn A function that accepts up to three arguments. The filter method calls the callbackfn function one time for each element in the array.
       * @param thisArg An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
       */
      filter: PropTypes.func.isRequired,
      /**
       * Returns the value of the first element in the array where predicate is true, and undefined
       * otherwise.
       * @param predicate find calls predicate once for each element of the array, in ascending
       * order, until it finds one where predicate returns true. If such an element is found, find
       * immediately returns that element value. Otherwise, find returns undefined.
       * @param thisArg If provided, it will be used as the this value for each invocation of
       * predicate. If it is not provided, undefined is used instead.
       */
      find: PropTypes.func.isRequired,
      /**
       * Returns the index of the first element in the array where predicate is true, and -1
       * otherwise.
       * @param predicate find calls predicate once for each element of the array, in ascending
       * order, until it finds one where predicate returns true. If such an element is found,
       * findIndex immediately returns that element index. Otherwise, findIndex returns -1.
       * @param thisArg If provided, it will be used as the this value for each invocation of
       * predicate. If it is not provided, undefined is used instead.
       */
      findIndex: PropTypes.func.isRequired,
      /**
       * Returns a new array with all sub-array elements concatenated into it recursively up to the
       * specified depth.
       *
       * @param depth The maximum recursion depth
       */
      flat: PropTypes.func.isRequired,
      /**
       * Calls a defined callback function on each element of an array. Then, flattens the result into
       * a new array.
       * This is identical to a map followed by flat with depth 1.
       *
       * @param callback A function that accepts up to three arguments. The flatMap method calls the
       * callback function one time for each element in the array.
       * @param thisArg An object to which the this keyword can refer in the callback function. If
       * thisArg is omitted, undefined is used as the this value.
       */
      flatMap: PropTypes.func.isRequired,
      /**
       * Performs the specified action for each element in an array.
       * @param callbackfn  A function that accepts up to three arguments. forEach calls the callbackfn function one time for each element in the array.
       * @param thisArg  An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
       */
      forEach: PropTypes.func.isRequired,
      /**
       * Determines whether an array includes a certain element, returning true or false as appropriate.
       * @param searchElement The element to search for.
       * @param fromIndex The position in this array at which to begin searching for searchElement.
       */
      includes: PropTypes.func.isRequired,
      /**
       * Returns the index of the first occurrence of a value in an array.
       * @param searchElement The value to locate in the array.
       * @param fromIndex The array index at which to begin the search. If fromIndex is omitted, the search starts at index 0.
       */
      indexOf: PropTypes.func.isRequired,
      /**
       * Adds all the elements of an array separated by the specified separator string.
       * @param separator A string used to separate one element of an array from the next in the resulting String. If omitted, the array elements are separated with a comma.
       */
      join: PropTypes.func.isRequired,
      /**
       * Returns an iterable of keys in the array
       */
      keys: PropTypes.func.isRequired,
      /**
       * Returns the index of the last occurrence of a specified value in an array.
       * @param searchElement The value to locate in the array.
       * @param fromIndex The array index at which to begin the search. If fromIndex is omitted, the search starts at the last index in the array.
       */
      lastIndexOf: PropTypes.func.isRequired,
      /**
       * Gets or sets the length of the array. This is a number one higher than the highest element defined in an array.
       */
      length: PropTypes.number.isRequired,
      /**
       * Calls a defined callback function on each element of an array, and returns an array that contains the results.
       * @param callbackfn A function that accepts up to three arguments. The map method calls the callbackfn function one time for each element in the array.
       * @param thisArg An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
       */
      map: PropTypes.func.isRequired,
      /**
       * Removes the last element from an array and returns it.
       */
      pop: PropTypes.func.isRequired,
      /**
       * Appends new elements to an array, and returns the new length of the array.
       * @param items New elements of the Array.
       */
      push: PropTypes.func.isRequired,
      /**
       * Calls the specified callback function for all the elements in an array. The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.
       * @param callbackfn A function that accepts up to four arguments. The reduce method calls the callbackfn function one time for each element in the array.
       * @param initialValue If initialValue is specified, it is used as the initial value to start the accumulation. The first call to the callbackfn function provides this value as an argument instead of an array value.
       */
      reduce: PropTypes.func.isRequired,
      /**
       * Calls the specified callback function for all the elements in an array, in descending order. The return value of the callback function is the accumulated result, and is provided as an argument in the next call to the callback function.
       * @param callbackfn A function that accepts up to four arguments. The reduceRight method calls the callbackfn function one time for each element in the array.
       * @param initialValue If initialValue is specified, it is used as the initial value to start the accumulation. The first call to the callbackfn function provides this value as an argument instead of an array value.
       */
      reduceRight: PropTypes.func.isRequired,
      /**
       * Reverses the elements in an Array.
       */
      reverse: PropTypes.func.isRequired,
      /**
       * Removes the first element from an array and returns it.
       */
      shift: PropTypes.func.isRequired,
      /**
       * Returns a section of an array.
       * @param start The beginning of the specified portion of the array.
       * @param end The end of the specified portion of the array. This is exclusive of the element at the index 'end'.
       */
      slice: PropTypes.func.isRequired,
      /**
       * Determines whether the specified callback function returns true for any element of an array.
       * @param callbackfn A function that accepts up to three arguments. The some method calls
       * the callbackfn function for each element in the array until the callbackfn returns a value
       * which is coercible to the Boolean value true, or until the end of the array.
       * @param thisArg An object to which the this keyword can refer in the callbackfn function.
       * If thisArg is omitted, undefined is used as the this value.
       */
      some: PropTypes.func.isRequired,
      /**
       * Sorts an array.
       * @param compareFn Function used to determine the order of the elements. It is expected to return
       * a negative value if first argument is less than second argument, zero if they're equal and a positive
       * value otherwise. If omitted, the elements are sorted in ascending, ASCII character order.
       * ```ts
       * [11,2,22,1].sort((a, b) => a - b)
       * ```
       */
      sort: PropTypes.func.isRequired,
      /**
       * Removes elements from an array and, if necessary, inserts new elements in their place, returning the deleted elements.
       * @param start The zero-based location in the array from which to start removing elements.
       * @param deleteCount The number of elements to remove.
       */
      splice: PropTypes.func.isRequired,
      /**
       * Returns a string containing a number represented in exponential notation.
       * @param fractionDigits Number of digits after the decimal point. Must be in the range 0 - 20, inclusive.
       */
      toExponential: PropTypes.func.isRequired,
      /**
       * Returns a string representing a number in fixed-point notation.
       * @param fractionDigits Number of digits after the decimal point. Must be in the range 0 - 20, inclusive.
       */
      toFixed: PropTypes.func.isRequired,
      /**
       * Returns a string representation of an array. The elements are converted to string using their toLocalString methods.
       */
      toLocaleString: PropTypes.func.isRequired,
      /**
       * Returns a string containing a number represented either in exponential or fixed-point notation with a specified number of digits.
       * @param precision Number of significant digits. Must be in the range 1 - 21, inclusive.
       */
      toPrecision: PropTypes.func.isRequired,
      /**
       * Returns a string representation of an array.
       */
      toString: PropTypes.func.isRequired,
      /**
       * Inserts new elements at the start of an array.
       * @param items  Elements to insert at the start of the Array.
       */
      unshift: PropTypes.func.isRequired,
      /**
       * Returns the primitive value of the specified object.
       */
      valueOf: PropTypes.func.isRequired,
      /**
       * Returns an iterable of values in the array
       */
      values: PropTypes.func.isRequired,
    }),
  ]).isRequired,
  /**
   * Задает полосу внизу панели
   *  @default true
   */
  classicBar: PropTypes.bool,
  /**
   * Задает дополнительные классы для элемента
   */
  className: PropTypes.string,
  /**
   * Задает disabled для всей группы
   */
  disabledGroup: PropTypes.bool,
  /**
   * Callback функция, вызываемая при изменении значения
   */
  onChange: PropTypes.func,
  /**
   * Задает прокрутку для табов.
   *  @param false Табы устанавливаются на всю ширину, если не помещаются - обрезаются многоточием
   *  @param true Табы занимают только ширину табов, оставляя пространство, если не помещаются - появляется прокрутка
   *  @default false
   */
  scrollable: PropTypes.bool,
  /**
   * Задает размер
   *  @default "medium"
   */
  size: PropTypes.oneOf(['medium', 'small']),
  /**
   * Задает дополнительные стили для компонента
   */
  style: PropTypes.object,
  /**
   * Задает выбор активного таба.
   *  Рекомендуется по дефолту выбирать первый tab
   */
  value: PropTypes.string.isRequired,
};

TabsClassicGroup.displayName = 'TabsClassicGroup';
exports.default = TabsClassicGroup;
