// rtk-ui-kit/components/PickerDate/PickerDate/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledPickerDateContainerTime =
  exports.StyledPickerDateContainerYear =
  exports.StyledPickerDateContainerMonth =
  exports.StyledPickerDateNavButton =
  exports.StyledPickerDateContainer =
  exports.StyledPickerDate =
    void 0;

var styled_components_1 = __importDefault(require('styled-components'));

var styledGenerators_1 = require('./styledGenerators');

var styledGenerators_2 = require('../styledGenerators');

var Typography_1 = __importDefault(require('../../Typography/Typography'));

var styles_1 = require('../CalendarTimes/styles');

exports.StyledPickerDate = styled_components_1.default.div(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        "\n  width: 280px;\n  font-family: 'Rostelecom Basis';\n  background-color: ",
        ';\n  box-shadow: ',
        ';\n  ',
        '\n\n  ',
        ' {\n    max-height: 220px;\n  }\n',
      ],
      [
        "\n  width: 280px;\n  font-family: 'Rostelecom Basis';\n  background-color: ",
        ';\n  box-shadow: ',
        ';\n  ',
        '\n\n  ',
        ' {\n    max-height: 220px;\n  }\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.dropdownMenu.background;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.shadows.bottomXL;
  },
  styledGenerators_1.generatePickerShape,
  styles_1.StyledCalendarTimesGrid
);
exports.StyledPickerDateContainer = styled_components_1.default.div(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      ['\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n'],
      ['\n  display: flex;\n  justify-content: space-between;\n  align-items: center;\n']
    ))
);
exports.StyledPickerDateNavButton = styled_components_1.default.div(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      [
        '\n  width: 36px;\n  height: 36px;\n  cursor: ',
        ";\n  border-radius: 8px;\n  margin: 12px;\n  font-feature-settings: 'tnum' on, 'lnum' on;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  ",
        ';\n  :hover {\n    background: ',
        ';\n  }\n',
      ],
      [
        '\n  width: 36px;\n  height: 36px;\n  cursor: ',
        ";\n  border-radius: 8px;\n  margin: 12px;\n  font-feature-settings: 'tnum' on, 'lnum' on;\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  ",
        ';\n  :hover {\n    background: ',
        ';\n  }\n',
      ]
    )),
  function (_a) {
    var isDayView = _a.isDayView;
    return isDayView ? 'pointer' : 'default';
  },
  styledGenerators_1.generatePickerShape,
  function (_a) {
    var theme = _a.theme,
      isDayView = _a.isDayView;
    return isDayView ? theme.transparent[5] : 'inherit';
  }
);
exports.StyledPickerDateContainerMonth = styled_components_1.default.div(
  templateObject_4 ||
    (templateObject_4 = __makeTemplateObject(
      [
        "\n  font-size: 16px;\n  line-height: 24px;\n  font-feature-settings: 'tnum' on, 'lnum' on;\n  cursor: pointer;\n  padding: 4px 8px;\n  ",
        ';\n  color: ',
        ';\n  background: ',
        ';\n  :hover {\n    background: ',
        ';\n  }\n',
      ],
      [
        "\n  font-size: 16px;\n  line-height: 24px;\n  font-feature-settings: 'tnum' on, 'lnum' on;\n  cursor: pointer;\n  padding: 4px 8px;\n  ",
        ';\n  color: ',
        ';\n  background: ',
        ';\n  :hover {\n    background: ',
        ';\n  }\n',
      ]
    )),
  styledGenerators_2.generateItemShape,
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[100];
  },
  function (_a) {
    var theme = _a.theme,
      active = _a.active;
    return active ? theme.transparent[5] : 'inherit';
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[5];
  }
);
exports.StyledPickerDateContainerYear = styled_components_1.default.div(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      [
        "\n  font-size: 16px;\n  line-height: 24px;\n  font-feature-settings: 'tnum' on, 'lnum' on;\n  cursor: pointer;\n  padding: 4px 10px;\n  ",
        ';\n  color: ',
        ';\n  background: ',
        ';\n  :hover {\n    background: ',
        ';\n    color: ',
        ';\n  }\n',
      ],
      [
        "\n  font-size: 16px;\n  line-height: 24px;\n  font-feature-settings: 'tnum' on, 'lnum' on;\n  cursor: pointer;\n  padding: 4px 10px;\n  ",
        ';\n  color: ',
        ';\n  background: ',
        ';\n  :hover {\n    background: ',
        ';\n    color: ',
        ';\n  }\n',
      ]
    )),
  styledGenerators_2.generateItemShape,
  function (_a) {
    var theme = _a.theme,
      active = _a.active;
    return active ? theme.transparent[100] : theme.transparent[50];
  },
  function (_a) {
    var theme = _a.theme,
      active = _a.active;
    return active ? theme.transparent[5] : 'inherit';
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[5];
  },
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[100];
  }
);
exports.StyledPickerDateContainerTime = styled_components_1.default(Typography_1.default)(
  templateObject_6 ||
    (templateObject_6 = __makeTemplateObject(
      ['\n  cursor: pointer;\n  padding: 4px 8px;\n  ', ';\n'],
      ['\n  cursor: pointer;\n  padding: 4px 8px;\n  ', ';\n']
    )),
  styledGenerators_2.generateItemShape
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6;
