// rtk-ui-kit/components/Grid/Col/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledCol = exports.StyledColSpacing = void 0;

var styled_components_1 = __importStar(require('styled-components'));

var components_1 = require('../../../constants/components');

var generateWidth = function generateWidth(size) {
  if (size) {
    var width = (100 / 12) * size;
    return styled_components_1.css(
      templateObject_1 ||
        (templateObject_1 = __makeTemplateObject(
          ['\n      flex: 0 0 ', '%;\n      max-width: ', '%;\n    '],
          ['\n      flex: 0 0 ', '%;\n      max-width: ', '%;\n    ']
        )),
      width,
      width
    );
  }

  return '';
};

var generateOffset = function generateOffset(offset) {
  if (offset === 0) {
    return styled_components_1.css(
      templateObject_2 ||
        (templateObject_2 = __makeTemplateObject(
          ['\n      margin-left: 0;\n    '],
          ['\n      margin-left: 0;\n    ']
        ))
    );
  }

  if (offset) {
    var gap = (100 / 12) * offset;
    return styled_components_1.css(
      templateObject_3 ||
        (templateObject_3 = __makeTemplateObject(
          ['\n      margin-left: ', '%;\n    '],
          ['\n      margin-left: ', '%;\n    ']
        )),
      gap
    );
  }

  return '';
};

var generateWidthMobileS = function generateWidthMobileS(_a) {
  var mobileS = _a.mobileS;
  return generateWidth(mobileS);
};

var generateWidthTabletS = function generateWidthTabletS(_a) {
  var tabletS = _a.tabletS;
  return generateWidth(tabletS);
};

var generateWidthTabletM = function generateWidthTabletM(_a) {
  var tabletM = _a.tabletM;
  return generateWidth(tabletM);
};

var generateWidthTabletL = function generateWidthTabletL(_a) {
  var tabletL = _a.tabletL;
  return generateWidth(tabletL);
};

var generateWidthDesktopS = function generateWidthDesktopS(_a) {
  var desktopS = _a.desktopS;
  return generateWidth(desktopS);
};

var generateWidthDesktopM = function generateWidthDesktopM(_a) {
  var desktopM = _a.desktopM;
  return generateWidth(desktopM);
};

var generateOffsetMobileS = function generateOffsetMobileS(_a) {
  var offsetMobileS = _a.offsetMobileS;
  return generateOffset(offsetMobileS);
};

var generateOffsetTabletS = function generateOffsetTabletS(_a) {
  var offsetTabletS = _a.offsetTabletS;
  return generateOffset(offsetTabletS);
};

var generateOffsetTabletM = function generateOffsetTabletM(_a) {
  var offsetTabletM = _a.offsetTabletM;
  return generateOffset(offsetTabletM);
};

var generateOffsetTabletL = function generateOffsetTabletL(_a) {
  var offsetTabletL = _a.offsetTabletL;
  return generateOffset(offsetTabletL);
};

var generateOffsetDesktopS = function generateOffsetDesktopS(_a) {
  var offsetDesktopS = _a.offsetDesktopS;
  return generateOffset(offsetDesktopS);
};

var generateOffsetDesktopM = function generateOffsetDesktopM(_a) {
  var offsetDesktopM = _a.offsetDesktopM;
  return generateOffset(offsetDesktopM);
};

exports.StyledColSpacing = styled_components_1.default.div(
  templateObject_4 ||
    (templateObject_4 = __makeTemplateObject(
      [
        '\n  ',
        ';\n  min-height: 1px;\n  padding-left: 8px;\n\n  ',
        ';\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    padding-left: 16px;\n\n    ',
        ';\n  }\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    ',
        ';\n  }\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    ',
        ';\n  }\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    ',
        ';\n  }\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    padding-left: 32px;\n\n    ',
        ';\n  }\n',
      ],
      [
        '\n  ',
        ';\n  min-height: 1px;\n  padding-left: 8px;\n\n  ',
        ';\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    padding-left: 16px;\n\n    ',
        ';\n  }\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    ',
        ';\n  }\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    ',
        ';\n  }\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    ',
        ';\n  }\n\n  @media screen and (min-width: ',
        'px) {\n    ',
        ';\n    padding-left: 32px;\n\n    ',
        ';\n  }\n',
      ]
    )),
  generateWidthMobileS,
  generateOffsetMobileS,
  components_1.BREAKPOINTS.tablet_vertical_small,
  generateWidthTabletS,
  generateOffsetTabletS,
  components_1.BREAKPOINTS.tablet_vertical_large,
  generateWidthTabletM,
  generateOffsetTabletM,
  components_1.BREAKPOINTS.tablet_horizontal,
  generateWidthTabletL,
  generateOffsetTabletL,
  components_1.BREAKPOINTS.desktop_small,
  generateWidthDesktopS,
  generateOffsetDesktopS,
  components_1.BREAKPOINTS.desktop_middle,
  generateWidthDesktopM,
  generateOffsetDesktopM
);
exports.StyledCol = styled_components_1.default.div(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      ['\n  width: 100%;\n  box-sizing: border-box;\n'],
      ['\n  width: 100%;\n  box-sizing: border-box;\n']
    ))
);
var templateObject_1, templateObject_2, templateObject_3, templateObject_4, templateObject_5;
