// rtk-ui-kit/components/Tag/TagItem/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledTagItem =
  exports.StyledTagItemButton =
  exports.StyledText =
  exports.generateTextColor =
    void 0;

var styled_components_1 = __importStar(require('styled-components'));

var styledGenerators_1 = require('../styledGenerators');

var BaseButton_1 = __importDefault(require('../../Base/BaseButton/BaseButton'));

var generateTextColor = function generateTextColor(_a) {
  var theme = _a.theme,
    disabled = _a.disabled;

  if (disabled) {
    return styled_components_1.css(
      templateObject_1 ||
        (templateObject_1 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.transparent[25]
    );
  }

  return styled_components_1.css(
    templateObject_2 ||
      (templateObject_2 = __makeTemplateObject(
        ['\n    color: ', ';\n  '],
        ['\n    color: ', ';\n  ']
      )),
    theme.transparent[100]
  );
};

exports.generateTextColor = generateTextColor;

var generateTagItemView = function generateTagItemView(_a) {
  var theme = _a.theme,
    disabled = _a.disabled;

  if (disabled) {
    return styled_components_1.css(
      templateObject_3 ||
        (templateObject_3 = __makeTemplateObject(
          ['\n      background-color: ', ';\n    '],
          ['\n      background-color: ', ';\n    ']
        )),
      theme.transparent[5]
    );
  }

  return styled_components_1.css(
    templateObject_4 ||
      (templateObject_4 = __makeTemplateObject(['\n    ', ';\n  '], ['\n    ', ';\n  '])),
    styledGenerators_1.generateTagView
  );
};

var generateTagItemButton = function generateTagItemButton(_a) {
  var theme = _a.theme,
    disabled = _a.disabled;

  if (disabled) {
    return styled_components_1.css(
      templateObject_5 ||
        (templateObject_5 = __makeTemplateObject(
          ['\n      cursor: not-allowed;\n\n      svg {\n        fill: ', ';\n      }\n    '],
          ['\n      cursor: not-allowed;\n\n      svg {\n        fill: ', ';\n      }\n    ']
        )),
      theme.transparent[25]
    );
  }

  return styled_components_1.css(
    templateObject_6 ||
      (templateObject_6 = __makeTemplateObject(
        [
          '\n    @media not all and (hover: none) {\n      &:hover svg {\n        fill: ',
          ';\n      }\n    }\n  ',
        ],
        [
          '\n    @media not all and (hover: none) {\n      &:hover svg {\n        fill: ',
          ';\n      }\n    }\n  ',
        ]
      )),
    theme.transparent[100]
  );
};

exports.StyledText = styled_components_1.default.span(
  templateObject_7 ||
    (templateObject_7 = __makeTemplateObject(
      [
        '\n  text-overflow: ellipsis;\n  overflow: hidden;\n  white-space: nowrap;\n\n  ',
        ';\n  ',
        ';\n',
      ],
      [
        '\n  text-overflow: ellipsis;\n  overflow: hidden;\n  white-space: nowrap;\n\n  ',
        ';\n  ',
        ';\n',
      ]
    )),
  exports.generateTextColor,
  styledGenerators_1.generateTextSize
);
exports.StyledTagItemButton = styled_components_1.default(BaseButton_1.default)(
  templateObject_8 ||
    (templateObject_8 = __makeTemplateObject(
      [
        '\n  background-color: transparent;\n  padding: 0;\n  text-decoration: none;\n  border-radius: inherit;\n  border: 0;\n  box-sizing: border-box;\n  outline: none;\n\n  display: inline-flex;\n  flex-shrink: 0;\n  cursor: pointer;\n\n  ',
        ';\n',
      ],
      [
        '\n  background-color: transparent;\n  padding: 0;\n  text-decoration: none;\n  border-radius: inherit;\n  border: 0;\n  box-sizing: border-box;\n  outline: none;\n\n  display: inline-flex;\n  flex-shrink: 0;\n  cursor: pointer;\n\n  ',
        ';\n',
      ]
    )),
  generateTagItemButton
);
exports.StyledTagItem = styled_components_1.default.div(
  templateObject_9 ||
    (templateObject_9 = __makeTemplateObject(
      [
        '\n  box-sizing: border-box;\n  display: inline-flex;\n  align-items: center;\n  max-width: 100%;\n\n  ',
        ';\n  ',
        '\n  ',
        ';\n  ',
        ';\n',
      ],
      [
        '\n  box-sizing: border-box;\n  display: inline-flex;\n  align-items: center;\n  max-width: 100%;\n\n  ',
        ';\n  ',
        '\n  ',
        ';\n  ',
        ';\n',
      ]
    )),
  styledGenerators_1.generateTagSize,
  function (_a) {
    var closable = _a.closable;
    return closable ? styledGenerators_1.generateTagRightPadding : null;
  },
  styledGenerators_1.generateTagShape,
  generateTagItemView
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9;
