// rtk-ui-kit/components/RadioButton/RadioButton/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledRadioButton = exports.StyledCheckmark = void 0;

var styled_components_1 = __importStar(require('styled-components'));

var generateDisabledStylesCheckmark = function generateDisabledStylesCheckmark(_a) {
  var disabled = _a.disabled;

  if (!disabled) {
    return styled_components_1.css(
      templateObject_1 ||
        (templateObject_1 = __makeTemplateObject(
          ['\n      cursor: pointer;\n    '],
          ['\n      cursor: pointer;\n    ']
        ))
    );
  }

  return styled_components_1.css(
    templateObject_2 ||
      (templateObject_2 = __makeTemplateObject(
        ['\n    color: ', ';\n    cursor: not-allowed;\n  '],
        ['\n    color: ', ';\n    cursor: not-allowed;\n  ']
      )),
    function (_a) {
      var theme = _a.theme;
      return theme.transparent[25];
    }
  );
};

var generateDisabledStylesCheckmarkBefore = function generateDisabledStylesCheckmarkBefore(_a) {
  var disabled = _a.disabled;

  if (disabled) {
    return styled_components_1.css(
      templateObject_3 ||
        (templateObject_3 = __makeTemplateObject(
          ['\n      background: ', ';\n      border: 1px solid transparent;\n    '],
          ['\n      background: ', ';\n      border: 1px solid transparent;\n    ']
        )),
      function (_a) {
        var theme = _a.theme;
        return theme.transparent[10];
      }
    );
  }

  return styled_components_1.css(
    templateObject_4 ||
      (templateObject_4 = __makeTemplateObject(
        ['\n    cursor: pointer;\n  '],
        ['\n    cursor: pointer;\n  ']
      ))
  );
};

var generateDisabledStylesCheckmarkHoverBefore =
  function generateDisabledStylesCheckmarkHoverBefore(_a) {
    var disabled = _a.disabled,
      color = _a.color;

    if (!disabled) {
      return styled_components_1.css(
        templateObject_5 ||
          (templateObject_5 = __makeTemplateObject(
            ['\n      border: 1px solid ', ';\n    '],
            ['\n      border: 1px solid ', ';\n    ']
          )),
        color
      );
    }

    return styled_components_1.css(
      templateObject_6 || (templateObject_6 = __makeTemplateObject([''], ['']))
    );
  };

var generateDisabledStylesRadioButtonBefore = function generateDisabledStylesRadioButtonBefore(_a) {
  var disabled = _a.disabled,
    theme = _a.theme,
    color = _a.color;

  if (disabled) {
    return styled_components_1.css(
      templateObject_7 ||
        (templateObject_7 = __makeTemplateObject(
          ['\n      background: ', ';\n      border: 1px solid transparent;\n    '],
          ['\n      background: ', ';\n      border: 1px solid transparent;\n    ']
        )),
      theme.transparent[10]
    );
  }

  return styled_components_1.css(
    templateObject_8 ||
      (templateObject_8 = __makeTemplateObject(
        ['\n    background: transparent;\n    border: 1px solid ', ';\n  '],
        ['\n    background: transparent;\n    border: 1px solid ', ';\n  ']
      )),
    color
  );
};

var generateDisabledStylesRadioButtonAfter = function generateDisabledStylesRadioButtonAfter(_a) {
  var disabled = _a.disabled,
    theme = _a.theme;

  if (disabled) {
    return styled_components_1.css(
      templateObject_9 ||
        (templateObject_9 = __makeTemplateObject(
          ['\n      background: ', ';\n    '],
          ['\n      background: ', ';\n    ']
        )),
      theme.transparent[25]
    );
  }

  return styled_components_1.css(
    templateObject_10 ||
      (templateObject_10 = __makeTemplateObject(
        ['\n    background: ', ';\n  '],
        ['\n    background: ', ';\n  ']
      )),
    theme.input.checkmark
  );
};

exports.StyledCheckmark = styled_components_1.default.span(
  templateObject_11 ||
    (templateObject_11 = __makeTemplateObject(
      [
        '\n  display: flex;\n  align-items: center;\n  margin-left: 32px;\n  ',
        "\n\n  &:before,\n  &:after {\n    box-sizing: border-box;\n    content: '';\n    position: absolute;\n    top: 1px;\n    left: 1px;\n  }\n\n  &:before {\n    width: 22px;\n    height: 22px;\n    background: transparent;\n    border: 1px solid ",
        ';\n    border-radius: 50%;\n    transition: border-color 0.1s linear;\n    ',
        '\n  }\n\n  &:hover:before {\n    ',
        '\n  }\n',
      ],
      [
        '\n  display: flex;\n  align-items: center;\n  margin-left: 32px;\n  ',
        "\n\n  &:before,\n  &:after {\n    box-sizing: border-box;\n    content: '';\n    position: absolute;\n    top: 1px;\n    left: 1px;\n  }\n\n  &:before {\n    width: 22px;\n    height: 22px;\n    background: transparent;\n    border: 1px solid ",
        ';\n    border-radius: 50%;\n    transition: border-color 0.1s linear;\n    ',
        '\n  }\n\n  &:hover:before {\n    ',
        '\n  }\n',
      ]
    )),
  generateDisabledStylesCheckmark,
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[25];
  },
  generateDisabledStylesCheckmarkBefore,
  generateDisabledStylesCheckmarkHoverBefore
);
exports.StyledRadioButton = styled_components_1.default.label(
  templateObject_12 ||
    (templateObject_12 = __makeTemplateObject(
      [
        '\n  -webkit-tap-highlight-color: transparent;\n  display: inline-flex;\n  align-items: center;\n  min-height: 24px;\n  position: relative;\n  outline: none;\n  > input {\n    position: absolute;\n    width: 0;\n    height: 0;\n    overflow: hidden;\n    opacity: 0;\n\n    &:checked + ',
        '::before {\n      ',
        '\n    }\n\n    &:checked + ',
        '::after {\n      width: 8px;\n      height: 8px;\n      border-radius: 50%;\n      transform: translate(7px, 7px);\n      ',
        '\n    }\n\n    &:focus:enabled + ',
        '::before {\n      border-width: 2px;\n    }\n  }\n',
      ],
      [
        '\n  -webkit-tap-highlight-color: transparent;\n  display: inline-flex;\n  align-items: center;\n  min-height: 24px;\n  position: relative;\n  outline: none;\n  > input {\n    position: absolute;\n    width: 0;\n    height: 0;\n    overflow: hidden;\n    opacity: 0;\n\n    &:checked + ',
        '::before {\n      ',
        '\n    }\n\n    &:checked + ',
        '::after {\n      width: 8px;\n      height: 8px;\n      border-radius: 50%;\n      transform: translate(7px, 7px);\n      ',
        '\n    }\n\n    &:focus:enabled + ',
        '::before {\n      border-width: 2px;\n    }\n  }\n',
      ]
    )),
  exports.StyledCheckmark,
  generateDisabledStylesRadioButtonBefore,
  exports.StyledCheckmark,
  generateDisabledStylesRadioButtonAfter,
  exports.StyledCheckmark
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12;
