// rtk-ui-kit/components/Button/Button/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledButton = exports.StyledText = void 0;

var styled_components_1 = __importStar(require('styled-components'));

var components_1 = require('../../../constants/components');

var Typography_1 = __importDefault(require('../../Typography/Typography'));

var BaseButton_1 = __importDefault(require('../../Base/BaseButton/BaseButton'));

exports.StyledText = styled_components_1.default(Typography_1.default)(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      ['\n  display: inline-block;\n  text-overflow: ellipsis;\n  overflow: hidden;\n'],
      ['\n  display: inline-block;\n  text-overflow: ellipsis;\n  overflow: hidden;\n']
    ))
);

var generateSize = function generateSize(_a) {
  var size = _a.size;

  switch (size) {
    case components_1.SIZES.small:
      return styled_components_1.css(
        templateObject_2 ||
          (templateObject_2 = __makeTemplateObject(
            ['\n        min-width: 80px;\n        height: 32px;\n        padding: 0 12px;\n      '],
            ['\n        min-width: 80px;\n        height: 32px;\n        padding: 0 12px;\n      ']
          ))
      );

    case components_1.SIZES.medium:
      return styled_components_1.css(
        templateObject_3 ||
          (templateObject_3 = __makeTemplateObject(
            [
              '\n        min-width: 120px;\n        height: 48px;\n        padding: 0 24px;\n      ',
            ],
            ['\n        min-width: 120px;\n        height: 48px;\n        padding: 0 24px;\n      ']
          ))
      );

    case components_1.SIZES.large:
      return styled_components_1.css(
        templateObject_4 ||
          (templateObject_4 = __makeTemplateObject(
            [
              '\n        min-width: 140px;\n        height: 56px;\n        padding: 0 40px;\n      ',
            ],
            ['\n        min-width: 140px;\n        height: 56px;\n        padding: 0 40px;\n      ']
          ))
      );

    default:
      return '';
  }
};

var generateView = function generateView(_a) {
  var view = _a.view,
    color = _a.color,
    colorhover = _a.colorhover,
    coloractive = _a.coloractive,
    theme = _a.theme;

  switch (view) {
    case components_1.BUTTON_VIEWS.primary:
      return styled_components_1.css(
        templateObject_5 ||
          (templateObject_5 = __makeTemplateObject(
            [
              '\n        background-color: ',
              ';\n\n        & > span {\n          color: ',
              ';\n        }\n\n        & svg {\n          fill: ',
              ';\n        }\n\n        @media not all and (hover: none) {\n          &:hover:enabled {\n            background-image: linear-gradient(',
              ', ',
              ');\n          }\n        }\n\n        &:active:enabled {\n          background-image: linear-gradient(',
              ', ',
              ');\n        }\n      ',
            ],
            [
              '\n        background-color: ',
              ';\n\n        & > span {\n          color: ',
              ';\n        }\n\n        & svg {\n          fill: ',
              ';\n        }\n\n        @media not all and (hover: none) {\n          &:hover:enabled {\n            background-image: linear-gradient(',
              ', ',
              ');\n          }\n        }\n\n        &:active:enabled {\n          background-image: linear-gradient(',
              ', ',
              ');\n        }\n      ',
            ]
          )),
        color,
        theme.greyScale.white,
        theme.greyScale.white,
        theme.transparent[5],
        theme.transparent[5],
        theme.transparent[10],
        theme.transparent[10]
      );

    case components_1.BUTTON_VIEWS.ghost:
      return styled_components_1.css(
        templateObject_6 ||
          (templateObject_6 = __makeTemplateObject(
            [
              '\n        & > span {\n          color: ',
              ';\n        }\n\n        & svg {\n          fill: ',
              ';\n        }\n\n        @media not all and (hover: none) {\n          &:hover:enabled {\n            background-color: ',
              ';\n          }\n        }\n\n        &:active:enabled {\n          background-color: ',
              ';\n        }\n      ',
            ],
            [
              '\n        & > span {\n          color: ',
              ';\n        }\n\n        & svg {\n          fill: ',
              ';\n        }\n\n        @media not all and (hover: none) {\n          &:hover:enabled {\n            background-color: ',
              ';\n          }\n        }\n\n        &:active:enabled {\n          background-color: ',
              ';\n        }\n      ',
            ]
          )),
        color,
        color,
        colorhover,
        coloractive
      );

    case components_1.BUTTON_VIEWS.outline:
      return styled_components_1.css(
        templateObject_7 ||
          (templateObject_7 = __makeTemplateObject(
            [
              '\n        border-color: ',
              ';\n\n        & > span {\n          color: ',
              ';\n        }\n\n        & svg {\n          fill: ',
              ';\n        }\n\n        @media not all and (hover: none) {\n          &:hover:enabled {\n            & > span {\n              color: ',
              ';\n            }\n\n            & svg {\n              fill: ',
              ';\n            }\n          }\n        }\n      ',
            ],
            [
              '\n        border-color: ',
              ';\n\n        & > span {\n          color: ',
              ';\n        }\n\n        & svg {\n          fill: ',
              ';\n        }\n\n        @media not all and (hover: none) {\n          &:hover:enabled {\n            & > span {\n              color: ',
              ';\n            }\n\n            & svg {\n              fill: ',
              ';\n            }\n          }\n        }\n      ',
            ]
          )),
        color,
        theme.transparent[100],
        theme.transparent[100],
        color,
        color
      );

    default:
      return '';
  }
};

var generateShape = function generateShape(_a) {
  var shape = _a.shape;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_8 ||
          (templateObject_8 = __makeTemplateObject(
            ['\n        border-radius: 0px;\n      '],
            ['\n        border-radius: 0px;\n      ']
          ))
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_9 ||
          (templateObject_9 = __makeTemplateObject(
            ['\n        border-radius: 8px;\n      '],
            ['\n        border-radius: 8px;\n      ']
          ))
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_10 ||
          (templateObject_10 = __makeTemplateObject(
            ['\n        border-radius: 32px;\n      '],
            ['\n        border-radius: 32px;\n      ']
          ))
      );

    default:
      return '';
  }
};

var generateDisabled = function generateDisabled(_a) {
  var disabled = _a.disabled,
    view = _a.view,
    theme = _a.theme;
  var baseDisabledStyles = styled_components_1.css(
    templateObject_11 ||
      (templateObject_11 = __makeTemplateObject(
        [
          '\n    cursor: not-allowed;\n\n    & > span {\n      color: ',
          ';\n    }\n\n    & svg {\n      fill: ',
          ';\n    }\n  ',
        ],
        [
          '\n    cursor: not-allowed;\n\n    & > span {\n      color: ',
          ';\n    }\n\n    & svg {\n      fill: ',
          ';\n    }\n  ',
        ]
      )),
    theme.transparent[25],
    theme.transparent[25]
  );

  if (disabled && view === components_1.BUTTON_VIEWS.primary) {
    return styled_components_1.css(
      templateObject_12 ||
        (templateObject_12 = __makeTemplateObject(
          ['\n      ', ';\n      background-color: ', ';\n    '],
          ['\n      ', ';\n      background-color: ', ';\n    ']
        )),
      baseDisabledStyles,
      theme.transparent[10]
    );
  }

  if (disabled && view === components_1.BUTTON_VIEWS.outline) {
    return styled_components_1.css(
      templateObject_13 ||
        (templateObject_13 = __makeTemplateObject(
          ['\n      ', ';\n      border-color: ', ';\n    '],
          ['\n      ', ';\n      border-color: ', ';\n    ']
        )),
      baseDisabledStyles,
      theme.transparent[10]
    );
  }

  if (disabled && view === components_1.BUTTON_VIEWS.ghost) {
    return styled_components_1.css(
      templateObject_14 ||
        (templateObject_14 = __makeTemplateObject(
          ['\n      ', ';\n    '],
          ['\n      ', ';\n    ']
        )),
      baseDisabledStyles
    );
  }

  return '';
};

exports.StyledButton = styled_components_1.default(BaseButton_1.default)(
  templateObject_15 ||
    (templateObject_15 = __makeTemplateObject(
      [
        '\n  & {\n    background-color: transparent;\n    padding: 0;\n    text-decoration: none;\n    border-radius: inherit;\n    border: 2px solid transparent;\n    box-sizing: border-box;\n    outline: none;\n\n    display: flex;\n    justify-content: center;\n    align-items: center;\n    cursor: pointer;\n    max-width: 100%;\n    white-space: nowrap;\n\n    ',
        ';\n    ',
        ';\n    ',
        ';\n    ',
        ';\n\n    svg {\n      flex-shrink: 0;\n    }\n\n    > span + svg,\n    > svg + span {\n      margin-left: 8px;\n    }\n  }\n',
      ],
      [
        '\n  & {\n    background-color: transparent;\n    padding: 0;\n    text-decoration: none;\n    border-radius: inherit;\n    border: 2px solid transparent;\n    box-sizing: border-box;\n    outline: none;\n\n    display: flex;\n    justify-content: center;\n    align-items: center;\n    cursor: pointer;\n    max-width: 100%;\n    white-space: nowrap;\n\n    ',
        ';\n    ',
        ';\n    ',
        ';\n    ',
        ';\n\n    svg {\n      flex-shrink: 0;\n    }\n\n    > span + svg,\n    > svg + span {\n      margin-left: 8px;\n    }\n  }\n',
      ]
    )),
  generateSize,
  generateView,
  generateShape,
  generateDisabled
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12,
  templateObject_13,
  templateObject_14,
  templateObject_15;
