// rtk-ui-kit/components/DropdownMenu/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledDropdownMenuOptionText =
  exports.StyledDropdownMenuOptionTextWrapper =
  exports.StyledDropdownMenuOptionIcon =
  exports.StyledDropdownMenuOption =
  exports.StyledDropdownMenuOptions =
  exports.StyledDropdownMenuHeader =
  exports.StyledDropdownMenuContainer =
  exports.StyledDropdownMenu =
    void 0;

var styled_components_1 = __importDefault(require('styled-components'));

var scroll_1 = require('../../styles/scroll');

var styledGenerators_1 = require('./styledGenerators');

var Typography_1 = __importDefault(require('../Typography/Typography'));

exports.StyledDropdownMenu = styled_components_1.default.div(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  ',
        ';\n  ',
        ';\n  mask-image: ',
        ';\n  box-sizing: border-box;\n  padding: 12px 0 12px 0;\n',
      ],
      [
        '\n  ',
        ';\n  ',
        ';\n  mask-image: ',
        ';\n  box-sizing: border-box;\n  padding: 12px 0 12px 0;\n',
      ]
    )),
  styledGenerators_1.generateMenuWidth,
  styledGenerators_1.generateShapeDropdown,
  function (_a) {
    var theme = _a.theme;
    return (
      'linear-gradient(\n    to bottom,\n    ' +
      theme.dropdownMenu.backgroundGradient +
      ',\n    ' +
      theme.dropdownMenu.background +
      ' 5% 95%,\n    ' +
      theme.dropdownMenu.backgroundGradient +
      '\n  )'
    );
  }
);
exports.StyledDropdownMenuContainer = styled_components_1.default.div(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      ['\n  overflow: auto;\n  max-height: 256px;\n\n  ', '\n'],
      ['\n  overflow: auto;\n  max-height: 256px;\n\n  ', '\n']
    )),
  scroll_1.scrollStyles
);
exports.StyledDropdownMenuHeader = styled_components_1.default(Typography_1.default)(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      ['\n  padding: 8px 20px;\n  word-break: normal;\n'],
      ['\n  padding: 8px 20px;\n  word-break: normal;\n']
    ))
);
exports.StyledDropdownMenuOptions = styled_components_1.default.ul(
  templateObject_4 ||
    (templateObject_4 = __makeTemplateObject(
      ['\n  padding-inline-start: 0;\n  margin-block-end: 0;\n  margin-block-start: 0;\n'],
      ['\n  padding-inline-start: 0;\n  margin-block-end: 0;\n  margin-block-start: 0;\n']
    ))
);
exports.StyledDropdownMenuOption = styled_components_1.default.li(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      [
        '\n  display: flex;\n  padding: 8px 20px;\n  cursor: pointer;\n\n  &:hover {\n    background-color: ',
        ';\n  }\n',
      ],
      [
        '\n  display: flex;\n  padding: 8px 20px;\n  cursor: pointer;\n\n  &:hover {\n    background-color: ',
        ';\n  }\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[5];
  }
);
exports.StyledDropdownMenuOptionIcon = styled_components_1.default.div(
  templateObject_6 ||
    (templateObject_6 = __makeTemplateObject(
      [
        '\n  display: flex;\n  align-items: center;\n  margin-right: 8px;\n\n  & svg {\n    ',
        ';\n  }\n',
      ],
      [
        '\n  display: flex;\n  align-items: center;\n  margin-right: 8px;\n\n  & svg {\n    ',
        ';\n  }\n',
      ]
    )),
  styledGenerators_1.generateFill
);
exports.StyledDropdownMenuOptionTextWrapper = styled_components_1.default.div(
  templateObject_7 || (templateObject_7 = __makeTemplateObject(['\n  ', ';\n'], ['\n  ', ';\n'])),
  styledGenerators_1.generateTextMargin
);
exports.StyledDropdownMenuOptionText = styled_components_1.default(Typography_1.default)(
  templateObject_8 ||
    (templateObject_8 = __makeTemplateObject(
      ['\n  ', ';\n\n  a {\n    color: inherit;\n    text-decoration: none;\n  }\n'],
      ['\n  ', ';\n\n  a {\n    color: inherit;\n    text-decoration: none;\n  }\n']
    )),
  styledGenerators_1.generateColor
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8;
