// rtk-ui-kit/components/Base/BaseInput/BaseInput.js
'use strict';
var PropTypes = require('prop-types');

var __assign =
  (void 0 && (void 0).__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];

          for (var p in s) {
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
          }
        }

        return t;
      };

    return __assign.apply(this, arguments);
  };

var __rest =
  (void 0 && (void 0).__rest) ||
  function (s, e) {
    var t = {};

    for (var p in s) {
      if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    }

    if (s != null && typeof Object.getOwnPropertySymbols === 'function')
      for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
          t[p[i]] = s[p[i]];
      }
    return t;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importDefault(require('react'));

var BaseInput = react_1.default.forwardRef(function BaseInput(props, ref) {
  var name = props.name,
    type = props.type,
    onChange = props.onChange,
    disabled = props.disabled,
    _a = props.value,
    value = _a === void 0 ? '' : _a,
    checked = props.checked,
    restProps = __rest(props, ['name', 'type', 'onChange', 'disabled', 'value', 'checked']);

  return react_1.default.createElement(
    'input',
    __assign(
      {
        name: name,
        type: type,
        onChange: onChange,
        disabled: disabled,
        ref: ref,
        value: value,
        checked: checked,
      },
      restProps
    )
  );
});

BaseInput.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  type: PropTypes.string,
  value: PropTypes.string,
};

exports.default = BaseInput;
