// rtk-ui-kit/components/Switch/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledText =
  exports.StyledInput =
  exports.StyledSlider =
  exports.StyledSliderHoverPlace =
  exports.StyledLabel =
  exports.StyledTouchContainer =
    void 0;

var styled_components_1 = __importDefault(require('styled-components'));

var BaseInput_1 = __importDefault(require('../Base/BaseInput/BaseInput'));

var constants_1 = require('./constants');

var styledGenerators_1 = require('./styledGenerators');

exports.StyledTouchContainer = styled_components_1.default.span(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      ['\n  display: inline-flex;\n  flex-wrap: nowrap;\n  align-items: flex-start;\n'],
      ['\n  display: inline-flex;\n  flex-wrap: nowrap;\n  align-items: flex-start;\n']
    ))
);
exports.StyledLabel = styled_components_1.default.label(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      ['\n  display: inline-flex;\n'],
      ['\n  display: inline-flex;\n']
    ))
);
exports.StyledSliderHoverPlace = styled_components_1.default.span(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      ['\n  width: 100%;\n  height: 100%;\n  position: absolute;\n  z-index: 1;\n'],
      ['\n  width: 100%;\n  height: 100%;\n  position: absolute;\n  z-index: 1;\n']
    ))
);
exports.StyledSlider = styled_components_1.default.span(
  templateObject_4 ||
    (templateObject_4 = __makeTemplateObject(
      [
        '\n  position: relative;\n  cursor: pointer;\n  top: 0;\n  left: 0;\n  right: 0;\n  bottom: 0;\n  min-width: 44px;\n  height: 24px;\n  background-color: ',
        ';\n  overflow: hidden;\n  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);\n  ',
        '\n  ',
        '\n  ',
        '\n \n  @media not all and (hover: none) {\n    &:hover > ',
        ' {\n      ',
        "\n    }\n  }\n\n  &:before {\n    position: absolute;\n    content: '';\n    height: 20px;\n    width: 20px;\n    left: 2px;\n    bottom: 2px;\n    background-color: ",
        ';\n    box-shadow: ',
        ';\n    transition: transform 0.3s cubic-bezier(0.18, 0.89, 0.35, 1.15);\n    z-index: 2;\n    ',
        '\n    ',
        '\n  }\n',
      ],
      [
        '\n  position: relative;\n  cursor: pointer;\n  top: 0;\n  left: 0;\n  right: 0;\n  bottom: 0;\n  min-width: 44px;\n  height: 24px;\n  background-color: ',
        ';\n  overflow: hidden;\n  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);\n  ',
        '\n  ',
        '\n  ',
        '\n \n  @media not all and (hover: none) {\n    &:hover > ',
        ' {\n      ',
        "\n    }\n  }\n\n  &:before {\n    position: absolute;\n    content: '';\n    height: 20px;\n    width: 20px;\n    left: 2px;\n    bottom: 2px;\n    background-color: ",
        ';\n    box-shadow: ',
        ';\n    transition: transform 0.3s cubic-bezier(0.18, 0.89, 0.35, 1.15);\n    z-index: 2;\n    ',
        '\n    ',
        '\n  }\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[25];
  },
  styledGenerators_1.generateSwitchShape,
  styledGenerators_1.generateSwitchDisabled,
  styledGenerators_1.generateSliderTransition,
  exports.StyledSliderHoverPlace,
  styledGenerators_1.generateSliderBackground,
  function (_a) {
    var theme = _a.theme;
    return theme.greyScale.white;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.shadows.control;
  },
  styledGenerators_1.generateKnobShape,
  styledGenerators_1.generateKnobDisabled
);
exports.StyledInput = styled_components_1.default(BaseInput_1.default)(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      [
        '\n  display: none;\n  opacity: 0;\n  width: 0;\n  height: 0;\n  margin: 0;\n\n  &:checked + ',
        ' {\n    ',
        '\n  }\n\n  &:checked + ',
        ':before {\n    transform: translateX(',
        'px);\n  }\n',
      ],
      [
        '\n  display: none;\n  opacity: 0;\n  width: 0;\n  height: 0;\n  margin: 0;\n\n  &:checked + ',
        ' {\n    ',
        '\n  }\n\n  &:checked + ',
        ':before {\n    transform: translateX(',
        'px);\n  }\n',
      ]
    )),
  exports.StyledSlider,
  styledGenerators_1.generateColor,
  exports.StyledSlider,
  constants_1.SWITCH_DISPLACEMENT
);
exports.StyledText = styled_components_1.default.span(
  templateObject_6 ||
    (templateObject_6 = __makeTemplateObject(
      ['\n  cursor: pointer;\n\n  ', '\n  ', '\n'],
      ['\n  cursor: pointer;\n\n  ', '\n  ', '\n']
    )),
  styledGenerators_1.generateTextPosition,
  styledGenerators_1.generateTextDisabled
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6;
