// rtk-ui-kit/components/Switch/styledGenerators.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var _a, _b, _c;

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.generateSliderTransition =
  exports.generateSliderBackground =
  exports.generateTextPosition =
  exports.generateColor =
  exports.generateTextDisabled =
  exports.generateKnobDisabled =
  exports.generateSwitchDisabled =
  exports.generateKnobShape =
  exports.generateSwitchShape =
    void 0;

var styled_components_1 = require('styled-components');

var components_1 = require('../../constants/components');

var themeUtils_1 = require('../../utils/themeUtils');

var SWITCH_SHAPES =
  ((_a = {}),
  (_a[components_1.SHAPES.geometric] = styled_components_1.css(
    templateObject_1 ||
      (templateObject_1 = __makeTemplateObject(
        ['\n    border-radius: 0;\n  '],
        ['\n    border-radius: 0;\n  ']
      ))
  )),
  (_a[components_1.SHAPES.rounded] = styled_components_1.css(
    templateObject_2 ||
      (templateObject_2 = __makeTemplateObject(
        ['\n    border-radius: 8px;\n  '],
        ['\n    border-radius: 8px;\n  ']
      ))
  )),
  (_a[components_1.SHAPES.circular] = styled_components_1.css(
    templateObject_3 ||
      (templateObject_3 = __makeTemplateObject(
        ['\n    border-radius: 24px;\n  '],
        ['\n    border-radius: 24px;\n  ']
      ))
  )),
  _a);
var KNOB_SHAPES =
  ((_b = {}),
  (_b[components_1.SHAPES.geometric] = styled_components_1.css(
    templateObject_4 ||
      (templateObject_4 = __makeTemplateObject(
        ['\n    border-radius: 0;\n  '],
        ['\n    border-radius: 0;\n  ']
      ))
  )),
  (_b[components_1.SHAPES.rounded] = styled_components_1.css(
    templateObject_5 ||
      (templateObject_5 = __makeTemplateObject(
        ['\n    border-radius: 6px;\n  '],
        ['\n    border-radius: 6px;\n  ']
      ))
  )),
  (_b[components_1.SHAPES.circular] = styled_components_1.css(
    templateObject_6 ||
      (templateObject_6 = __makeTemplateObject(
        ['\n    border-radius: 14px;\n  '],
        ['\n    border-radius: 14px;\n  ']
      ))
  )),
  _b);
var TEXT_POSITION =
  ((_c = {}),
  (_c[components_1.TEXT_HORIZONTAL_POSITIONS.left] = styled_components_1.css(
    templateObject_7 ||
      (templateObject_7 = __makeTemplateObject(
        ['\n    padding-right: 16px;\n    order: -1;\n  '],
        ['\n    padding-right: 16px;\n    order: -1;\n  ']
      ))
  )),
  (_c[components_1.TEXT_HORIZONTAL_POSITIONS.right] = styled_components_1.css(
    templateObject_8 ||
      (templateObject_8 = __makeTemplateObject(
        ['\n    padding-left: 16px;\n    order: 1;\n  '],
        ['\n    padding-left: 16px;\n    order: 1;\n  ']
      ))
  )),
  _c);

var generateSwitchShape = function generateSwitchShape(_a) {
  var shape = _a.shape;
  return SWITCH_SHAPES[shape];
};

exports.generateSwitchShape = generateSwitchShape;

var generateKnobShape = function generateKnobShape(_a) {
  var shape = _a.shape;
  return KNOB_SHAPES[shape];
};

exports.generateKnobShape = generateKnobShape;

var generateSwitchDisabled = function generateSwitchDisabled(_a) {
  var disabled = _a.disabled,
    checked = _a.checked;

  if (disabled && checked) {
    return styled_components_1.css(
      templateObject_9 ||
        (templateObject_9 = __makeTemplateObject(
          ['\n      cursor: not-allowed;\n      opacity: 0.5;\n    '],
          ['\n      cursor: not-allowed;\n      opacity: 0.5;\n    ']
        ))
    );
  }

  if (disabled) {
    return styled_components_1.css(
      templateObject_10 ||
        (templateObject_10 = __makeTemplateObject(
          ['\n      cursor: not-allowed;\n      background-color: ', ';\n    '],
          ['\n      cursor: not-allowed;\n      background-color: ', ';\n    ']
        )),
      function (_a) {
        var theme = _a.theme;
        return theme.transparent[10];
      }
    );
  }

  return '';
};

exports.generateSwitchDisabled = generateSwitchDisabled;

var generateKnobDisabled = function generateKnobDisabled(_a) {
  var disabled = _a.disabled,
    checked = _a.checked;
  return disabled && !checked
    ? styled_components_1.css(
        templateObject_11 ||
          (templateObject_11 = __makeTemplateObject(
            ['\n        opacity: 0.75;\n      '],
            ['\n        opacity: 0.75;\n      ']
          ))
      )
    : styled_components_1.css(
        templateObject_12 || (templateObject_12 = __makeTemplateObject([''], ['']))
      );
};

exports.generateKnobDisabled = generateKnobDisabled;

var generateTextDisabled = function generateTextDisabled(_a) {
  var disabled = _a.disabled;
  return disabled
    ? styled_components_1.css(
        templateObject_13 ||
          (templateObject_13 = __makeTemplateObject(
            ['\n        cursor: default;\n      '],
            ['\n        cursor: default;\n      ']
          ))
      )
    : '';
};

exports.generateTextDisabled = generateTextDisabled;

var generateColor = function generateColor(_a) {
  var color = _a.color;
  return styled_components_1.css(
    templateObject_14 ||
      (templateObject_14 = __makeTemplateObject(
        ['\n  background-color: ', ';\n'],
        ['\n  background-color: ', ';\n']
      )),
    color
  );
};

exports.generateColor = generateColor;

var generateTextPosition = function generateTextPosition(_a) {
  var textPosition = _a.textPosition;
  return TEXT_POSITION[textPosition];
};

exports.generateTextPosition = generateTextPosition;

var generateSliderBackground = function generateSliderBackground(_a) {
  var theme = _a.theme,
    disabled = _a.disabled;
  return (
    !disabled &&
    styled_components_1.css(
      templateObject_15 ||
        (templateObject_15 = __makeTemplateObject(
          ['\n    background-color: ', ';\n  '],
          ['\n    background-color: ', ';\n  ']
        )),
      theme.transparent[5]
    )
  );
};

exports.generateSliderBackground = generateSliderBackground;

var generateSliderTransition = function generateSliderTransition(_a) {
  var theme = _a.theme;
  return themeUtils_1.generateThemeTransition(['background-color'], theme.transitions.easeInOut);
};

exports.generateSliderTransition = generateSliderTransition;
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12,
  templateObject_13,
  templateObject_14,
  templateObject_15;
