// rtk-ui-kit/components/InputNumberStepper/constants.js
'use strict';

var _a, _b, _c;

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.BTN_PADDING_LIST =
  exports.STEPPER_HEIGHT_LIST =
  exports.STEPPER_WIDTH_LIST =
  exports.MAX_LENGTH_FOR_CORRECT_DISPLAY_VAL =
  exports.NUMBER_STEPPER_VAL_MAX_LENGTH =
  exports.DEFAULT_COLOR =
  exports.DEFAULT_DISABLED_STATE =
  exports.DEFAULT_STEP =
  exports.DEFAULT_VALUE =
  exports.DEFAULT_SIZE =
    void 0;

var components_1 = require('../../constants/components');

var stepper_1 = require('../../constants/stepper');

exports.DEFAULT_SIZE = stepper_1.STEPPER_SIZES.medium;
exports.DEFAULT_VALUE = 0;
exports.DEFAULT_STEP = 1;
exports.DEFAULT_DISABLED_STATE = {
  isAllDisabled: false,
  isLeftDisabled: false,
  isRightDisabled: false,
};

exports.DEFAULT_COLOR = components_1.BASE_COLORS.primary2;
exports.NUMBER_STEPPER_VAL_MAX_LENGTH = 20;
exports.MAX_LENGTH_FOR_CORRECT_DISPLAY_VAL = 13;
exports.STEPPER_WIDTH_LIST =
  ((_a = {}),
  (_a[stepper_1.STEPPER_SIZES.medium] = 120),
  (_a[stepper_1.STEPPER_SIZES.small] = 80),
  _a);
exports.STEPPER_HEIGHT_LIST =
  ((_b = {}),
  (_b[stepper_1.STEPPER_SIZES.medium] = 48),
  (_b[stepper_1.STEPPER_SIZES.small] = 32),
  _b);
exports.BTN_PADDING_LIST =
  ((_c = {}),
  (_c[stepper_1.STEPPER_SIZES.medium] = 12),
  (_c[stepper_1.STEPPER_SIZES.small] = 4),
  _c);
