// rtk-ui-kit/components/InputNumberStepper/styledGenerators.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.generateCaretColor =
  exports.generateHintPadding =
  exports.generateHintColor =
  exports.generateInputDisabled =
  exports.generateInputFont =
  exports.generateContainerGeomerticBottomLine =
  exports.generateBtnBorderRadius =
  exports.generateBtnPadding =
  exports.generateHoveredBtn =
  exports.generateDisabledBtn =
  exports.generateStepperSize =
  exports.generateInlineFlex =
  exports.generateShape =
  exports.generateColor =
    void 0;

var styled_components_1 = require('styled-components');

var components_1 = require('../../constants/components');

var stepper_1 = require('../../constants/stepper');

var constants_1 = require('./constants');

var generateColor = function generateColor(_a) {
  var _b = _a.theme,
    numberStepper = _b.numberStepper,
    transparent = _b.transparent,
    disabled = _a.disabled;
  var color = numberStepper.textColor;

  if (disabled) {
    color = transparent['25'];
  }

  return color;
};

exports.generateColor = generateColor;

var generateShape = function generateShape(_a) {
  var _b;

  var shape = _a.shape,
    theme = _a.theme;
  var NumberStepperShapesList =
    ((_b = {}),
    (_b[components_1.SHAPES.geometric] = styled_components_1.css(
      templateObject_1 ||
        (templateObject_1 = __makeTemplateObject(
          ['\n      border-radius: 0;\n      background-color: none;\n    '],
          ['\n      border-radius: 0;\n      background-color: none;\n    ']
        ))
    )),
    (_b[components_1.SHAPES.rounded] = styled_components_1.css(
      templateObject_2 ||
        (templateObject_2 = __makeTemplateObject(
          ['\n      border-radius: 8px;\n      background-color: ', ';\n    '],
          ['\n      border-radius: 8px;\n      background-color: ', ';\n    ']
        )),
      theme.transparent[5]
    )),
    (_b[components_1.SHAPES.circular] = styled_components_1.css(
      templateObject_3 ||
        (templateObject_3 = __makeTemplateObject(
          ['\n      border-radius: 24px;\n      background-color: ', ';\n    '],
          ['\n      border-radius: 24px;\n      background-color: ', ';\n    ']
        )),
      theme.transparent[5]
    )),
    _b);
  return NumberStepperShapesList[shape];
};

exports.generateShape = generateShape;

var generateInlineFlex = function generateInlineFlex() {
  return styled_components_1.css(
    templateObject_4 ||
      (templateObject_4 = __makeTemplateObject(
        ['\n  display: inline-flex;\n  align-items: center;\n  justify-content: center;\n'],
        ['\n  display: inline-flex;\n  align-items: center;\n  justify-content: center;\n']
      ))
  );
};

exports.generateInlineFlex = generateInlineFlex;

var generateStepperSize = function generateStepperSize(_a) {
  var size = _a.size;
  var width = constants_1.STEPPER_WIDTH_LIST[size];
  var height = constants_1.STEPPER_HEIGHT_LIST[size];
  return width && height
    ? styled_components_1.css(
        templateObject_5 ||
          (templateObject_5 = __makeTemplateObject(
            ['\n        min-width: ', 'px;\n        height: ', 'px;\n      '],
            ['\n        min-width: ', 'px;\n        height: ', 'px;\n      ']
          )),
        width,
        height
      )
    : '';
};

exports.generateStepperSize = generateStepperSize;

var generateDisabledBtn = function generateDisabledBtn(_a) {
  var disabled = _a.disabled;
  return disabled
    ? styled_components_1.css(
        templateObject_6 ||
          (templateObject_6 = __makeTemplateObject(
            ['\n        cursor: not-allowed;\n        pointer-events: none;\n      '],
            ['\n        cursor: not-allowed;\n        pointer-events: none;\n      ']
          ))
      )
    : '';
};

exports.generateDisabledBtn = generateDisabledBtn;

var generateHoveredBtn = function generateHoveredBtn(_a) {
  var color = _a.color,
    disabled = _a.disabled;
  return disabled
    ? ''
    : styled_components_1.css(
        templateObject_7 ||
          (templateObject_7 = __makeTemplateObject(
            ['\n        & svg {\n          fill: ', ';\n        }\n      '],
            ['\n        & svg {\n          fill: ', ';\n        }\n      ']
          )),
        color
      );
};

exports.generateHoveredBtn = generateHoveredBtn;

var generateBtnPadding = function generateBtnPadding(_a) {
  var size = _a.size,
    shape = _a.shape,
    type = _a.type;
  var result = styled_components_1.css(
    templateObject_8 || (templateObject_8 = __makeTemplateObject([''], ['']))
  );

  if (shape === components_1.SHAPES.geometric) {
    return result;
  }

  if (type === 'left') {
    result = styled_components_1.css(
      templateObject_9 ||
        (templateObject_9 = __makeTemplateObject(
          ['\n      padding-left: ', 'px;\n    '],
          ['\n      padding-left: ', 'px;\n    ']
        )),
      constants_1.BTN_PADDING_LIST[size]
    );
  }

  if (type === 'right') {
    result = styled_components_1.css(
      templateObject_10 ||
        (templateObject_10 = __makeTemplateObject(
          ['\n      padding-right: ', 'px;\n    '],
          ['\n      padding-right: ', 'px;\n    ']
        )),
      constants_1.BTN_PADDING_LIST[size]
    );
  }

  return result;
};

exports.generateBtnPadding = generateBtnPadding;

var generateBtnBorderRadius = function generateBtnBorderRadius(_a) {
  var shape = _a.shape;
  return styled_components_1.css(
    templateObject_11 ||
      (templateObject_11 = __makeTemplateObject(
        ['\n    border-radius: ', ';\n  '],
        ['\n    border-radius: ', ';\n  ']
      )),
    shape !== components_1.SHAPES.geometric ? '4px' : 0
  );
};

exports.generateBtnBorderRadius = generateBtnBorderRadius;

var generateContainerGeomerticBottomLine = function generateContainerGeomerticBottomLine(_a) {
  var error = _a.error,
    shape = _a.shape,
    theme = _a.theme,
    isFocused = _a.isFocused,
    disabled = _a.disabled,
    color = _a.color;

  if (shape !== components_1.SHAPES.geometric) {
    return '';
  }

  var backgroundColor = theme.transparent[10];

  if (isFocused && color && !disabled) {
    backgroundColor = color;
  }

  if (error) {
    backgroundColor = theme.systemColors.error(1);
  }

  return styled_components_1.css(
    templateObject_12 ||
      (templateObject_12 = __makeTemplateObject(
        [
          "\n    &:after {\n      position: absolute;\n      content: '';\n      height: 2px;\n      width: 100%;\n      bottom: 0;\n      background-color: ",
          ';\n    }\n  ',
        ],
        [
          "\n    &:after {\n      position: absolute;\n      content: '';\n      height: 2px;\n      width: 100%;\n      bottom: 0;\n      background-color: ",
          ';\n    }\n  ',
        ]
      )),
    backgroundColor
  );
};

exports.generateContainerGeomerticBottomLine = generateContainerGeomerticBottomLine;

var generateInputFont = function generateInputFont(_a) {
  var $stepperSize = _a.$stepperSize,
    theme = _a.theme;
  return styled_components_1.css(
    templateObject_13 || (templateObject_13 = __makeTemplateObject(['\n  ', '\n'], ['\n  ', '\n'])),
    $stepperSize === stepper_1.STEPPER_SIZES.small
      ? theme.fonts.static.accentS
      : theme.fonts.static.accentM
  );
};

exports.generateInputFont = generateInputFont;

var generateInputDisabled = function generateInputDisabled(_a) {
  var disabled = _a.disabled;
  return disabled
    ? styled_components_1.css(
        templateObject_14 ||
          (templateObject_14 = __makeTemplateObject(
            ['\n        cursor: not-allowed;\n      '],
            ['\n        cursor: not-allowed;\n      ']
          ))
      )
    : '';
};

exports.generateInputDisabled = generateInputDisabled;

var generateHintColor = function generateHintColor(_a) {
  var disabled = _a.disabled,
    error = _a.error,
    theme = _a.theme;

  if (error) {
    return styled_components_1.css(
      templateObject_15 ||
        (templateObject_15 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.input.error
    );
  }

  if (disabled) {
    return styled_components_1.css(
      templateObject_16 ||
        (templateObject_16 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.transparent[25]
    );
  }

  return styled_components_1.css(
    templateObject_17 ||
      (templateObject_17 = __makeTemplateObject(
        ['\n    color: ', ';\n  '],
        ['\n    color: ', ';\n  ']
      )),
    theme.transparent[50]
  );
};

exports.generateHintColor = generateHintColor;

var generateHintPadding = function generateHintPadding(_a) {
  var shape = _a.shape;

  if (shape === components_1.SHAPES.circular) {
    return styled_components_1.css(
      templateObject_18 ||
        (templateObject_18 = __makeTemplateObject(
          ['\n      padding-left: 8px;\n      padding-right: 8px;\n    '],
          ['\n      padding-left: 8px;\n      padding-right: 8px;\n    ']
        ))
    );
  }

  return '';
};

exports.generateHintPadding = generateHintPadding;

var generateCaretColor = function generateCaretColor(_a) {
  var shape = _a.shape,
    color = _a.color;
  return styled_components_1.css(
    templateObject_19 ||
      (templateObject_19 = __makeTemplateObject(
        ['\n  caret-color: ', ';\n'],
        ['\n  caret-color: ', ';\n']
      )),
    shape !== components_1.SHAPES.geometric && color ? color : 'auto'
  );
};

exports.generateCaretColor = generateCaretColor;
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12,
  templateObject_13,
  templateObject_14,
  templateObject_15,
  templateObject_16,
  templateObject_17,
  templateObject_18,
  templateObject_19;
