// rtk-ui-kit/components/Chip/Chip/Chip.js
'use strict';
var PropTypes = require('prop-types');

var __assign =
  (void 0 && (void 0).__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];

          for (var p in s) {
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
          }
        }

        return t;
      };

    return __assign.apply(this, arguments);
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var __rest =
  (void 0 && (void 0).__rest) ||
  function (s, e) {
    var t = {};

    for (var p in s) {
      if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    }

    if (s != null && typeof Object.getOwnPropertySymbols === 'function')
      for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
          t[p[i]] = s[p[i]];
      }
    return t;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importStar(require('react'));

var context_1 = require('../context');

var useGetBaseColorFromProps_1 = require('../../../hooks/useGetBaseColorFromProps');

var useThemeContext_1 = require('../../../hooks/useThemeContext');

var components_1 = require('../../../constants/components');

var constants_1 = require('../constants');

var styles_1 = require('./styles');

var function_1 = require('../../../utils/function');

var Chip = function Chip(props) {
  var themeConfig = useThemeContext_1.useThemeContext().themeConfig;

  var _a = props.color,
    color = _a === void 0 ? constants_1.DEFAULT_CHIP_COLOR : _a,
    _b = props.shape,
    shape = _b === void 0 ? themeConfig.shape : _b,
    _c = props.view,
    view = _c === void 0 ? constants_1.DEFAULT_CHIP_VIEW : _c,
    amount = props.amount,
    children = props.children,
    _d = props.selected,
    selected = _d === void 0 ? false : _d,
    _e = props.disabled,
    disabled = _e === void 0 ? false : _e,
    id = props.id,
    className = props.className,
    style = props.style,
    name = props.name,
    _f = props.onClick,
    onClick = _f === void 0 ? function_1.noop : _f,
    restProps = __rest(props, [
      'color',
      'shape',
      'view',
      'amount',
      'children',
      'selected',
      'disabled',
      'id',
      'className',
      'style',
      'name',
      'onClick',
    ]);

  var baseColorFromProps = useGetBaseColorFromProps_1.useGetBaseColorFromProps(color);

  var _g = react_1.useContext(context_1.ChipGroupContext),
    groupSelected = _g.groupSelected,
    groupDisabled = _g.groupDisabled,
    onGroupChange = _g.onGroupChange;

  var _h = react_1.useState(selected),
    isSelected = _h[0],
    setIsSelected = _h[1];

  var _j = react_1.useState(disabled),
    isDisabled = _j[0],
    setIsDisabled = _j[1];

  var clickHandler = react_1.useCallback(
    function (evt) {
      setIsSelected(function (prevIsSelected) {
        return !prevIsSelected;
      });

      if (onGroupChange && id) {
        onGroupChange(id);
      }

      onClick(evt);
    },
    [id, onClick, onGroupChange]
  );
  react_1.useEffect(
    function () {
      if (groupSelected && id && !groupDisabled.includes(id)) {
        var newSelected = groupSelected.includes(id);
        setIsSelected(newSelected);
      } else if (selected !== isSelected && !disabled) {
        setIsSelected(selected);
      } // eslint-disable-next-line react-hooks/exhaustive-deps
    },
    [groupSelected, selected]
  );
  react_1.useEffect(
    function () {
      if (groupDisabled && id) {
        var newDisabled = groupDisabled.includes(id);
        setIsDisabled(newDisabled);
      } else if (disabled !== isDisabled) {
        setIsDisabled(disabled);
      } // eslint-disable-next-line react-hooks/exhaustive-deps
    },
    [groupDisabled, disabled]
  );
  return react_1.default.createElement(
    styles_1.StyledChip,
    __assign(
      {
        id: id,
        className: className,
        style: style,
        color: baseColorFromProps,
        selected: isSelected,
        disabled: isDisabled,
        shape: shape,
        view: view,
        type: 'button',
        onClick: clickHandler,
      },
      restProps
    ),
    react_1.default.createElement(
      styles_1.StyledText,
      {
        selected: isSelected,
        disabled: isDisabled,
        variant: components_1.TYPOGRAPHY_VARIANTS.bodyL,
        tag: 'span',
        isStatic: true,
      },
      children
    ),
    amount &&
      react_1.default.createElement(
        styles_1.StyledAmount,
        {
          selected: isSelected,
          disabled: isDisabled,
          variant: components_1.TYPOGRAPHY_VARIANTS.caption,
          tag: 'span',
          isStatic: true,
        },
        amount
      )
  );
};

Chip.propTypes = {
  /**
   * Задает значение счетчика
   */
  amount: PropTypes.number,
  children: PropTypes.node,
  /**
   * Задает дополнительные классы для компонента
   */
  className: PropTypes.string,
  /**
   * Задает основной цвет
   *  @default "primary2"
   */
  color: PropTypes.oneOf(['primary1', 'primary2', 'secondary1', 'secondary2']),
  /**
   * Устанавливает атрибут disabled
   */
  disabled: PropTypes.bool,
  /**
   * Задает id для компонента
   */
  id: PropTypes.string,
  /**
   * Задает атрибут name для компонента
   */
  name: PropTypes.string,
  /**
   * Callback функция, вызываемая при клике
   */
  onClick: PropTypes.func,
  /**
   * Ярлык активного чипа
   */
  selected: PropTypes.bool,
  /**
   * Задает скругление, значение по умолчанию берется из темы
   */
  shape: PropTypes.oneOf(['circular', 'geometric', 'rounded']),
  /**
   * Задает дополнительные стили для компонента
   */
  style: PropTypes.object,
  /**
   * Задает внешний вид
   *  @default "primary"
   */
  view: PropTypes.oneOf(['primary', 'secondary']),
};

Chip.displayName = 'Chip';
exports.default = Chip;
