// rtk-ui-kit/components/TabsClassic/TabsClassicItem/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledText = exports.StyledTabsClassicItem = void 0;

var styled_components_1 = __importStar(require('styled-components'));

var constants_1 = require('../constants');

var Typography_1 = __importDefault(require('../../Typography/Typography'));

var generateTabItemSize = function generateTabItemSize(_a) {
  var size = _a.size,
    iconPosition = _a.iconPosition;

  switch (size) {
    case constants_1.TAB_SIZES.small:
      return styled_components_1.css(
        templateObject_1 ||
          (templateObject_1 = __makeTemplateObject(
            ['\n        height: ', ';\n      '],
            ['\n        height: ', ';\n      ']
          )),
        iconPosition === constants_1.TABS_ICON_POSITIONS.top ? '56px' : '44px'
      );

    case constants_1.TAB_SIZES.medium:
      return styled_components_1.css(
        templateObject_2 ||
          (templateObject_2 = __makeTemplateObject(
            ['\n        height: 64px;\n      '],
            ['\n        height: 64px;\n      ']
          ))
      );

    default:
      return '';
  }
};

var generateTabItemIconSpace = function generateTabItemIconSpace(_a) {
  var iconPosition = _a.iconPosition;

  switch (iconPosition) {
    case constants_1.TABS_ICON_POSITIONS.top:
      return styled_components_1.css(
        templateObject_3 ||
          (templateObject_3 = __makeTemplateObject(
            [
              '\n        flex-direction: column;\n        justify-content: flex-start;\n\n        svg {\n          margin-bottom: 4px;\n        }\n      ',
            ],
            [
              '\n        flex-direction: column;\n        justify-content: flex-start;\n\n        svg {\n          margin-bottom: 4px;\n        }\n      ',
            ]
          ))
      );

    case constants_1.TABS_ICON_POSITIONS.left:
      return styled_components_1.css(
        templateObject_4 ||
          (templateObject_4 = __makeTemplateObject(
            ['\n        svg {\n          margin-right: 8px;\n        }\n      '],
            ['\n        svg {\n          margin-right: 8px;\n        }\n      ']
          ))
      );

    case constants_1.TABS_ICON_POSITIONS.right:
      return styled_components_1.css(
        templateObject_5 ||
          (templateObject_5 = __makeTemplateObject(
            [
              '\n        flex-direction: row-reverse;\n\n        svg {\n          margin-left: 8px;\n        }\n      ',
            ],
            [
              '\n        flex-direction: row-reverse;\n\n        svg {\n          margin-left: 8px;\n        }\n      ',
            ]
          ))
      );

    default:
      return '';
  }
};

var generateTabItemActive = function generateTabItemActive(_a) {
  var isActive = _a.isActive,
    accentColor = _a.accentColor,
    disabled = _a.disabled,
    theme = _a.theme;

  if (isActive && !disabled) {
    return styled_components_1.css(
      templateObject_6 ||
        (templateObject_6 = __makeTemplateObject(
          [
            '\n      color: ',
            ';\n\n      &:after {\n        width: 100%;\n        background-color: ',
            ';\n      }\n\n      svg {\n        fill: ',
            ';\n      }\n    ',
          ],
          [
            '\n      color: ',
            ';\n\n      &:after {\n        width: 100%;\n        background-color: ',
            ';\n      }\n\n      svg {\n        fill: ',
            ';\n      }\n    ',
          ]
        )),
      accentColor,
      accentColor,
      accentColor
    );
  }

  if (disabled) {
    var activeStyles = isActive
      ? styled_components_1.css(
          templateObject_7 ||
            (templateObject_7 = __makeTemplateObject(
              [
                '\n          &:after {\n            width: 100%;\n            background-color: ',
                ';\n          }\n        ',
              ],
              [
                '\n          &:after {\n            width: 100%;\n            background-color: ',
                ';\n          }\n        ',
              ]
            )),
          theme.transparent[25]
        )
      : null;
    return styled_components_1.css(
      templateObject_8 ||
        (templateObject_8 = __makeTemplateObject(
          [
            '\n      color: ',
            ';\n      cursor: not-allowed;\n\n      ',
            ';\n\n      svg {\n        fill: ',
            ';\n      }\n    ',
          ],
          [
            '\n      color: ',
            ';\n      cursor: not-allowed;\n\n      ',
            ';\n\n      svg {\n        fill: ',
            ';\n      }\n    ',
          ]
        )),
      theme.transparent[25],
      activeStyles,
      theme.transparent[25]
    );
  }

  return styled_components_1.css(
    templateObject_9 ||
      (templateObject_9 = __makeTemplateObject(
        [
          '\n    color: ',
          ';\n\n    svg {\n      fill: ',
          ';\n    }\n\n    @media not all and (hover: none) {\n      &:hover {\n        color: ',
          ';\n\n        svg {\n          fill: ',
          ';\n        }\n      }\n    }\n  ',
        ],
        [
          '\n    color: ',
          ';\n\n    svg {\n      fill: ',
          ';\n    }\n\n    @media not all and (hover: none) {\n      &:hover {\n        color: ',
          ';\n\n        svg {\n          fill: ',
          ';\n        }\n      }\n    }\n  ',
        ]
      )),
    theme.transparent[100],
    theme.transparent[100],
    accentColor,
    accentColor
  );
};

var generateTabItemScrollable = function generateTabItemScrollable(_a) {
  var scrollable = _a.scrollable;

  if (scrollable) {
    return styled_components_1.css(
      templateObject_10 || (templateObject_10 = __makeTemplateObject([''], ['']))
    );
  }

  return styled_components_1.css(
    templateObject_11 ||
      (templateObject_11 = __makeTemplateObject(
        ['\n    min-width: 0;\n    flex-basis: 0;\n    flex-grow: 1;\n  '],
        ['\n    min-width: 0;\n    flex-basis: 0;\n    flex-grow: 1;\n  ']
      ))
  );
};

var generateTabItemTextScrollable = function generateTabItemTextScrollable(_a) {
  var scrollable = _a.scrollable;

  if (scrollable) {
    return styled_components_1.css(
      templateObject_12 || (templateObject_12 = __makeTemplateObject([''], ['']))
    );
  }

  return styled_components_1.css(
    templateObject_13 ||
      (templateObject_13 = __makeTemplateObject(
        ['\n    text-overflow: ellipsis;\n    overflow: hidden;\n  '],
        ['\n    text-overflow: ellipsis;\n    overflow: hidden;\n  ']
      ))
  );
};

exports.StyledTabsClassicItem = styled_components_1.default.button(
  templateObject_14 ||
    (templateObject_14 = __makeTemplateObject(
      [
        "\n  font: inherit;\n  background-color: transparent;\n  border: 0;\n  padding: 0;\n  outline: none;\n\n  box-sizing: border-box;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  white-space: nowrap;\n  position: relative;\n  cursor: pointer;\n\n  svg {\n    flex-shrink: 0;\n  }\n\n  &:after {\n    display: block;\n    content: '';\n    position: absolute;\n    left: 0;\n    bottom: 0;\n    width: 0%;\n    height: 2px;\n    transition: width 0.2s ease 0s, background-color 0.2s ease 0s;\n    z-index: 2;\n  }\n\n  ",
        ';\n  ',
        ';\n  ',
        ';\n  ',
        ';\n',
      ],
      [
        "\n  font: inherit;\n  background-color: transparent;\n  border: 0;\n  padding: 0;\n  outline: none;\n\n  box-sizing: border-box;\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  white-space: nowrap;\n  position: relative;\n  cursor: pointer;\n\n  svg {\n    flex-shrink: 0;\n  }\n\n  &:after {\n    display: block;\n    content: '';\n    position: absolute;\n    left: 0;\n    bottom: 0;\n    width: 0%;\n    height: 2px;\n    transition: width 0.2s ease 0s, background-color 0.2s ease 0s;\n    z-index: 2;\n  }\n\n  ",
        ';\n  ',
        ';\n  ',
        ';\n  ',
        ';\n',
      ]
    )),
  generateTabItemSize,
  generateTabItemActive,
  generateTabItemIconSpace,
  generateTabItemScrollable
);
exports.StyledText = styled_components_1.default(Typography_1.default)(
  templateObject_15 ||
    (templateObject_15 = __makeTemplateObject(
      ['\n  display: inline-block;\n  color: inherit;\n\n  ', ';\n'],
      ['\n  display: inline-block;\n  color: inherit;\n\n  ', ';\n']
    )),
  generateTabItemTextScrollable
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12,
  templateObject_13,
  templateObject_14,
  templateObject_15;
