// rtk-ui-kit/components/TextArea/styledGenerators.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.generateTextAreaStyle =
  exports.generateTextareaScrollPadding =
  exports.generateHintFontColor =
  exports.generateLabelFontColor =
  exports.generateLabelFontSize =
  exports.generateTextareaView =
  exports.generateTextareaContainerShape =
  exports.generateTextareaGeometricLine =
  exports.generateGap =
  exports.generateTextDisabled =
    void 0;

var styled_components_1 = require('styled-components');

var components_1 = require('../../constants/components');

var scroll_1 = require('../../styles/scroll');

var generateTextDisabled = function generateTextDisabled(_a) {
  var disabled = _a.disabled,
    theme = _a.theme;

  if (disabled) {
    return styled_components_1.css(
      templateObject_1 ||
        (templateObject_1 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.transparent[25]
    );
  }

  return '';
};

exports.generateTextDisabled = generateTextDisabled;

var generateGap = function generateGap(_a) {
  var shape = _a.shape;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_2 ||
          (templateObject_2 = __makeTemplateObject(
            ['\n        padding: 0;\n      '],
            ['\n        padding: 0;\n      ']
          ))
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_3 ||
          (templateObject_3 = __makeTemplateObject(
            ['\n        padding: 0 12px;\n      '],
            ['\n        padding: 0 12px;\n      ']
          ))
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_4 ||
          (templateObject_4 = __makeTemplateObject(
            ['\n        padding: 0 20px;\n      '],
            ['\n        padding: 0 20px;\n      ']
          ))
      );

    default:
      return '';
  }
};

exports.generateGap = generateGap;

var generateTextareaGeometricLine = function generateTextareaGeometricLine(_a) {
  var theme = _a.theme,
    color = _a.color,
    error = _a.error,
    isFocused = _a.isFocused;
  return styled_components_1.css(
    templateObject_5 ||
      (templateObject_5 = __makeTemplateObject(
        [
          "\n    &:before {\n      content: '';\n      position: absolute;\n      bottom: -2px;\n      left: 0;\n      width: ",
          ';\n      height: 2px;\n      background-color: ',
          ";\n      transition: width 0.2s ease, background-color 0.2s ease;\n      z-index: 2;\n    }\n\n    &:after {\n      content: '';\n      position: absolute;\n      left: 0;\n      bottom: -2px;\n      width: 100%;\n      height: 2px;\n      background-color: ",
          ';\n      z-index: 1;\n    }\n  ',
        ],
        [
          "\n    &:before {\n      content: '';\n      position: absolute;\n      bottom: -2px;\n      left: 0;\n      width: ",
          ';\n      height: 2px;\n      background-color: ',
          ";\n      transition: width 0.2s ease, background-color 0.2s ease;\n      z-index: 2;\n    }\n\n    &:after {\n      content: '';\n      position: absolute;\n      left: 0;\n      bottom: -2px;\n      width: 100%;\n      height: 2px;\n      background-color: ",
          ';\n      z-index: 1;\n    }\n  ',
        ]
      )),
    isFocused ? '100%' : '0%',
    error ? theme.systemColors.error(1) : color,
    error ? theme.systemColors.error(1) : theme.transparent[10]
  );
};

exports.generateTextareaGeometricLine = generateTextareaGeometricLine;

var generateTextareaContainerShape = function generateTextareaContainerShape(_a) {
  var shape = _a.shape,
    theme = _a.theme;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_6 ||
          (templateObject_6 = __makeTemplateObject(
            [
              '\n        background-color: transparent;\n        border-radius: 0;\n\n        ',
              ';\n      ',
            ],
            [
              '\n        background-color: transparent;\n        border-radius: 0;\n\n        ',
              ';\n      ',
            ]
          )),
        exports.generateTextareaGeometricLine
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_7 ||
          (templateObject_7 = __makeTemplateObject(
            ['\n        background-color: ', ';\n        border-radius: 8px;\n      '],
            ['\n        background-color: ', ';\n        border-radius: 8px;\n      ']
          )),
        theme.transparent[5]
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_8 ||
          (templateObject_8 = __makeTemplateObject(
            ['\n        background-color: ', ';\n        border-radius: 24px;\n      '],
            ['\n        background-color: ', ';\n        border-radius: 24px;\n      ']
          )),
        theme.transparent[5]
      );

    default:
      return '';
  }
};

exports.generateTextareaContainerShape = generateTextareaContainerShape;

var generateTextareaView = function generateTextareaView(_a) {
  var disabled = _a.disabled,
    theme = _a.theme;

  if (disabled) {
    return styled_components_1.css(
      templateObject_9 ||
        (templateObject_9 = __makeTemplateObject(
          ['\n      ', ';\n\n      cursor: not-allowed;\n    '],
          ['\n      ', ';\n\n      cursor: not-allowed;\n    ']
        )),
      exports.generateTextDisabled
    );
  }

  return styled_components_1.css(
    templateObject_10 ||
      (templateObject_10 = __makeTemplateObject(
        ['\n    color: ', ';\n  '],
        ['\n    color: ', ';\n  ']
      )),
    theme.transparent[100]
  );
};

exports.generateTextareaView = generateTextareaView;

var generateLabelFontSize = function generateLabelFontSize(_a) {
  var value = _a.value,
    isFocused = _a.isFocused,
    theme = _a.theme;

  if (value || isFocused) {
    return styled_components_1.css(
      templateObject_11 ||
        (templateObject_11 = __makeTemplateObject(
          ['\n      ', ';\n      top: 4px;\n    '],
          ['\n      ', ';\n      top: 4px;\n    ']
        )),
      theme.fonts.static.caption
    );
  }

  return styled_components_1.css(
    templateObject_12 ||
      (templateObject_12 = __makeTemplateObject(
        ['\n    ', ';\n    top: 12px;\n  '],
        ['\n    ', ';\n    top: 12px;\n  ']
      )),
    theme.fonts.static.bodyL
  );
};

exports.generateLabelFontSize = generateLabelFontSize;

var generateLabelFontColor = function generateLabelFontColor(_a) {
  var value = _a.value,
    disabled = _a.disabled,
    isFocused = _a.isFocused,
    error = _a.error,
    color = _a.color,
    theme = _a.theme;

  if (disabled) {
    return styled_components_1.css(
      templateObject_13 ||
        (templateObject_13 = __makeTemplateObject(
          ['\n      ', ';\n    '],
          ['\n      ', ';\n    ']
        )),
      exports.generateTextDisabled
    );
  }

  if (error) {
    return styled_components_1.css(
      templateObject_14 ||
        (templateObject_14 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.input.error
    );
  }

  if (isFocused) {
    return styled_components_1.css(
      templateObject_15 ||
        (templateObject_15 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      color
    );
  }

  if (value) {
    return styled_components_1.css(
      templateObject_16 ||
        (templateObject_16 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.transparent[50]
    );
  }

  return styled_components_1.css(
    templateObject_17 ||
      (templateObject_17 = __makeTemplateObject(
        ['\n    color: ', ';\n  '],
        ['\n    color: ', ';\n  ']
      )),
    theme.transparent[75]
  );
};

exports.generateLabelFontColor = generateLabelFontColor;

var generateHintFontColor = function generateHintFontColor(_a) {
  var error = _a.error,
    disabled = _a.disabled,
    theme = _a.theme;

  if (error) {
    return styled_components_1.css(
      templateObject_18 ||
        (templateObject_18 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.input.error
    );
  }

  if (disabled) {
    return styled_components_1.css(
      templateObject_19 ||
        (templateObject_19 = __makeTemplateObject(
          ['\n      ', ';\n    '],
          ['\n      ', ';\n    ']
        )),
      exports.generateTextDisabled
    );
  }

  return '';
};

exports.generateHintFontColor = generateHintFontColor;

var generateTextareaScrollPadding = function generateTextareaScrollPadding(_a) {
  var shape = _a.shape;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_20 ||
          (templateObject_20 = __makeTemplateObject(
            ['\n        padding-right: 0;\n      '],
            ['\n        padding-right: 0;\n      ']
          ))
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_21 ||
          (templateObject_21 = __makeTemplateObject(
            ['\n        padding-right: 8px;\n      '],
            ['\n        padding-right: 8px;\n      ']
          ))
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_22 ||
          (templateObject_22 = __makeTemplateObject(
            ['\n        padding-right: 12px;\n      '],
            ['\n        padding-right: 12px;\n      ']
          ))
      );

    default:
      return '';
  }
};

exports.generateTextareaScrollPadding = generateTextareaScrollPadding;

var generateTextAreaStyle = function generateTextAreaStyle(_a) {
  var theme = _a.theme;
  return styled_components_1.css(
    templateObject_23 ||
      (templateObject_23 = __makeTemplateObject(
        [
          '\n    background-color: transparent;\n    border: none;\n    outline: none;\n    overflow: auto;\n    resize: none;\n\n    width: 100%;\n    ',
          ';\n    caret-color: ',
          ';\n\n    ',
          ';\n    ',
          ';\n    ',
          '\n  ',
        ],
        [
          '\n    background-color: transparent;\n    border: none;\n    outline: none;\n    overflow: auto;\n    resize: none;\n\n    width: 100%;\n    ',
          ';\n    caret-color: ',
          ';\n\n    ',
          ';\n    ',
          ';\n    ',
          '\n  ',
        ]
      )),
    theme.fonts.static.bodyL,
    theme.transparent[100],
    exports.generateTextareaView,
    exports.generateGap,
    scroll_1.scrollStyles
  );
};

exports.generateTextAreaStyle = generateTextAreaStyle;
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12,
  templateObject_13,
  templateObject_14,
  templateObject_15,
  templateObject_16,
  templateObject_17,
  templateObject_18,
  templateObject_19,
  templateObject_20,
  templateObject_21,
  templateObject_22,
  templateObject_23;
