// rtk-ui-kit/components/TextArea/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledTextLength =
  exports.StyledHint =
  exports.StyledFooter =
  exports.StyledLabel =
  exports.StyledTextareaAutosize =
  exports.StyledTextarea =
  exports.StyledContainer =
  exports.StyledRoot =
    void 0;

var styled_components_1 = __importDefault(require('styled-components'));

var react_textarea_autosize_1 = __importDefault(require('react-textarea-autosize'));

var styledGenerators_1 = require('./styledGenerators');

var Typography_1 = __importDefault(require('../Typography/Typography'));

exports.StyledRoot = styled_components_1.default.div(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      ['\n  display: inline-block;\n  min-width: 280px;\n  position: relative;\n'],
      ['\n  display: inline-block;\n  min-width: 280px;\n  position: relative;\n']
    ))
);
exports.StyledContainer = styled_components_1.default.div(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      [
        '\n  line-height: 0;\n  position: relative;\n  ',
        ';\n\n  box-sizing: border-box;\n  padding: 24px 8px 12px 0;\n  ',
        '\n',
      ],
      [
        '\n  line-height: 0;\n  position: relative;\n  ',
        ';\n\n  box-sizing: border-box;\n  padding: 24px 8px 12px 0;\n  ',
        '\n',
      ]
    )),
  styledGenerators_1.generateTextareaContainerShape,
  styledGenerators_1.generateTextareaScrollPadding
);
exports.StyledTextarea = styled_components_1.default.textarea(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      ['\n  min-height: 76px;\n  ', ';\n'],
      ['\n  min-height: 76px;\n  ', ';\n']
    )),
  styledGenerators_1.generateTextAreaStyle
);
exports.StyledTextareaAutosize = styled_components_1.default(react_textarea_autosize_1.default)(
  templateObject_4 || (templateObject_4 = __makeTemplateObject(['\n  ', ';\n'], ['\n  ', ';\n'])),
  styledGenerators_1.generateTextAreaStyle
);
exports.StyledLabel = styled_components_1.default.label(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      [
        '\n  width: 100%;\n  position: absolute;\n  left: 0;\n  white-space: nowrap;\n  text-overflow: ellipsis;\n  overflow: hidden;\n  pointer-events: none;\n  transition-property: background, font-size, top, color;\n  transition-timing-function: ease-out;\n  transition-duration: 0.15s;\n\n  ',
        ';\n  ',
        ';\n  ',
        ';\n',
      ],
      [
        '\n  width: 100%;\n  position: absolute;\n  left: 0;\n  white-space: nowrap;\n  text-overflow: ellipsis;\n  overflow: hidden;\n  pointer-events: none;\n  transition-property: background, font-size, top, color;\n  transition-timing-function: ease-out;\n  transition-duration: 0.15s;\n\n  ',
        ';\n  ',
        ';\n  ',
        ';\n',
      ]
    )),
  styledGenerators_1.generateLabelFontSize,
  styledGenerators_1.generateLabelFontColor,
  styledGenerators_1.generateGap
);
exports.StyledFooter = styled_components_1.default.div(
  templateObject_6 ||
    (templateObject_6 = __makeTemplateObject(
      ['\n  display: flex;\n  justify-content: space-between;\n  margin-top: 8px;\n\n  ', ';\n'],
      ['\n  display: flex;\n  justify-content: space-between;\n  margin-top: 8px;\n\n  ', ';\n']
    )),
  styledGenerators_1.generateGap
);
exports.StyledHint = styled_components_1.default(Typography_1.default)(
  templateObject_7 ||
    (templateObject_7 = __makeTemplateObject(
      ['\n  word-break: break-word;\n\n  ', ';\n'],
      ['\n  word-break: break-word;\n\n  ', ';\n']
    )),
  styledGenerators_1.generateHintFontColor
);
exports.StyledTextLength = styled_components_1.default(Typography_1.default)(
  templateObject_8 ||
    (templateObject_8 = __makeTemplateObject(
      ['\n  flex-shrink: 0;\n  margin-left: 10px;\n\n  ', ';\n'],
      ['\n  flex-shrink: 0;\n  margin-left: 10px;\n\n  ', ';\n']
    )),
  styledGenerators_1.generateTextDisabled
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8;
