// rtk-ui-kit/components/Input/styledGenerators.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.generateIconColor =
  exports.generateIconButtonPosition =
  exports.generateLabelFontColor =
  exports.generateLabelFontSize =
  exports.generateInputColor =
  exports.generateInputView =
  exports.generateInputGeometricLine =
  exports.generateEllipsisContent =
  exports.generateGapRight =
  exports.generateGap =
  exports.generateTextDisabled =
    void 0;

var styled_components_1 = require('styled-components');

var components_1 = require('../../constants/components');

var types_1 = require('./types');

var generateTextDisabled = function generateTextDisabled(_a) {
  var disabled = _a.disabled,
    theme = _a.theme;

  if (disabled) {
    return styled_components_1.css(
      templateObject_1 ||
        (templateObject_1 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.transparent[25]
    );
  }

  return '';
};

exports.generateTextDisabled = generateTextDisabled;

var generateGap = function generateGap(_a) {
  var shape = _a.shape;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_2 ||
          (templateObject_2 = __makeTemplateObject(
            ['\n        padding: 0;\n      '],
            ['\n        padding: 0;\n      ']
          ))
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_3 ||
          (templateObject_3 = __makeTemplateObject(
            ['\n        padding: 0 12px;\n      '],
            ['\n        padding: 0 12px;\n      ']
          ))
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_4 ||
          (templateObject_4 = __makeTemplateObject(
            ['\n        padding: 0 20px;\n      '],
            ['\n        padding: 0 20px;\n      ']
          ))
      );

    default:
      return styled_components_1.css(
        templateObject_5 || (templateObject_5 = __makeTemplateObject([''], ['']))
      );
  }
};

exports.generateGap = generateGap;

var generateGapRight = function generateGapRight(_a) {
  var $icon = _a.$icon,
    shape = _a.shape;

  if ($icon) {
    switch (shape) {
      case components_1.SHAPES.geometric:
        return styled_components_1.css(
          templateObject_6 ||
            (templateObject_6 = __makeTemplateObject(
              ['\n          padding-right: 24px;\n        '],
              ['\n          padding-right: 24px;\n        ']
            ))
        );

      case components_1.SHAPES.rounded:
        return styled_components_1.css(
          templateObject_7 ||
            (templateObject_7 = __makeTemplateObject(
              ['\n          padding-right: 32px;\n        '],
              ['\n          padding-right: 32px;\n        ']
            ))
        );

      case components_1.SHAPES.circular:
        return styled_components_1.css(
          templateObject_8 ||
            (templateObject_8 = __makeTemplateObject(
              ['\n          padding-right: 36px;\n        '],
              ['\n          padding-right: 36px;\n        ']
            ))
        );

      default:
        return '';
    }
  }

  return '';
};

exports.generateGapRight = generateGapRight;

var generateEllipsisContent = function generateEllipsisContent() {
  return styled_components_1.css(
    templateObject_9 ||
      (templateObject_9 = __makeTemplateObject(
        ['\n    text-overflow: ellipsis;\n    white-space: nowrap;\n    overflow: hidden;\n  '],
        ['\n    text-overflow: ellipsis;\n    white-space: nowrap;\n    overflow: hidden;\n  ']
      ))
  );
};

exports.generateEllipsisContent = generateEllipsisContent;

var generateInputGeometricLine = function generateInputGeometricLine(_a) {
  var theme = _a.theme,
    shape = _a.shape,
    error = _a.error,
    color = _a.color,
    isFocused = _a.isFocused;
  return (
    shape === components_1.SHAPES.geometric &&
    styled_components_1.css(
      templateObject_10 ||
        (templateObject_10 = __makeTemplateObject(
          [
            "\n      &::after {\n        content: '';\n        position: absolute;\n        bottom: -2px;\n        left: 0;\n        width: 100%;\n        height: 2px;\n        background-color: ",
            ";\n        z-index: 1;\n      }\n\n      &::before {\n        content: '';\n        position: absolute;\n        bottom: -2px;\n        left: 0;\n        width: ",
            ';\n        height: 2px;\n        transition: width 0.2s ease, background-color 0.2s ease;\n        background-color: ',
            ';\n        z-index: 2;\n      }\n    ',
          ],
          [
            "\n      &::after {\n        content: '';\n        position: absolute;\n        bottom: -2px;\n        left: 0;\n        width: 100%;\n        height: 2px;\n        background-color: ",
            ";\n        z-index: 1;\n      }\n\n      &::before {\n        content: '';\n        position: absolute;\n        bottom: -2px;\n        left: 0;\n        width: ",
            ';\n        height: 2px;\n        transition: width 0.2s ease, background-color 0.2s ease;\n        background-color: ',
            ';\n        z-index: 2;\n      }\n    ',
          ]
        )),
      error ? theme.systemColors.error(1) : theme.transparent[10],
      isFocused ? '100%' : '0%',
      error ? theme.systemColors.error(1) : color
    )
  );
};

exports.generateInputGeometricLine = generateInputGeometricLine;

var generateInputView = function generateInputView(_a) {
  var shape = _a.shape,
    $icon = _a.$icon,
    theme = _a.theme;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_11 ||
          (templateObject_11 = __makeTemplateObject(
            [
              '\n        background-color: transparent;\n        padding: 20px ',
              ' 4px 0;\n        border-radius: 0px;\n      ',
            ],
            [
              '\n        background-color: transparent;\n        padding: 20px ',
              ' 4px 0;\n        border-radius: 0px;\n      ',
            ]
          )),
        $icon ? '24px' : '0'
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_12 ||
          (templateObject_12 = __makeTemplateObject(
            [
              '\n        background-color: ',
              ';\n        padding: 20px ',
              ' 4px 12px;\n        border-radius: 8px;\n      ',
            ],
            [
              '\n        background-color: ',
              ';\n        padding: 20px ',
              ' 4px 12px;\n        border-radius: 8px;\n      ',
            ]
          )),
        theme.transparent[5],
        $icon ? '32px' : '12px'
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_13 ||
          (templateObject_13 = __makeTemplateObject(
            [
              '\n        background-color: ',
              ';\n        padding: 20px ',
              ' 4px 20px;\n        border-radius: 24px;\n      ',
            ],
            [
              '\n        background-color: ',
              ';\n        padding: 20px ',
              ' 4px 20px;\n        border-radius: 24px;\n      ',
            ]
          )),
        theme.transparent[5],
        $icon ? '36px' : '20px'
      );

    default:
      return styled_components_1.css(
        templateObject_14 || (templateObject_14 = __makeTemplateObject([''], ['']))
      );
  }
};

exports.generateInputView = generateInputView;

var generateInputColor = function generateInputColor(_a) {
  var disabled = _a.disabled,
    theme = _a.theme;

  if (disabled) {
    return styled_components_1.css(
      templateObject_15 ||
        (templateObject_15 = __makeTemplateObject(
          ['\n      ', ';\n\n      cursor: not-allowed;\n    '],
          ['\n      ', ';\n\n      cursor: not-allowed;\n    ']
        )),
      exports.generateTextDisabled
    );
  }

  return styled_components_1.css(
    templateObject_16 ||
      (templateObject_16 = __makeTemplateObject(
        ['\n    color: ', ';\n  '],
        ['\n    color: ', ';\n  ']
      )),
    theme.transparent[100]
  );
};

exports.generateInputColor = generateInputColor;

var generateLabelFontSize = function generateLabelFontSize(_a) {
  var value = _a.value,
    isFocused = _a.isFocused,
    theme = _a.theme,
    floatLabel = _a.floatLabel;

  if (value || isFocused || floatLabel === types_1.FLOAT_LABEL.always) {
    return styled_components_1.css(
      templateObject_17 ||
        (templateObject_17 = __makeTemplateObject(
          ['\n      ', ';\n      transform: translate(0, 4px);\n    '],
          ['\n      ', ';\n      transform: translate(0, 4px);\n    ']
        )),
      theme.fonts.static.caption
    );
  }

  return styled_components_1.css(
    templateObject_18 ||
      (templateObject_18 = __makeTemplateObject(
        ['\n    ', ';\n    transform: translate(0, 12px);\n  '],
        ['\n    ', ';\n    transform: translate(0, 12px);\n  ']
      )),
    theme.fonts.static.bodyL
  );
};

exports.generateLabelFontSize = generateLabelFontSize;

var generateLabelFontColor = function generateLabelFontColor(_a) {
  var value = _a.value,
    disabled = _a.disabled,
    isFocused = _a.isFocused,
    error = _a.error,
    color = _a.color,
    theme = _a.theme,
    floatLabel = _a.floatLabel;

  if (disabled) {
    return styled_components_1.css(
      templateObject_19 ||
        (templateObject_19 = __makeTemplateObject(
          ['\n      ', ';\n    '],
          ['\n      ', ';\n    ']
        )),
      exports.generateTextDisabled
    );
  }

  if (error) {
    return styled_components_1.css(
      templateObject_20 ||
        (templateObject_20 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.input.error
    );
  }

  if (isFocused) {
    return styled_components_1.css(
      templateObject_21 ||
        (templateObject_21 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      color
    );
  }

  if (value || floatLabel === types_1.FLOAT_LABEL.always) {
    return styled_components_1.css(
      templateObject_22 ||
        (templateObject_22 = __makeTemplateObject(
          ['\n      color: ', ';\n    '],
          ['\n      color: ', ';\n    ']
        )),
      theme.transparent[50]
    );
  }

  return styled_components_1.css(
    templateObject_23 ||
      (templateObject_23 = __makeTemplateObject(
        ['\n    color: ', ';\n  '],
        ['\n    color: ', ';\n  ']
      )),
    theme.transparent[75]
  );
};

exports.generateLabelFontColor = generateLabelFontColor;

var generateIconButtonPosition = function generateIconButtonPosition(_a) {
  var shape = _a.shape;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_24 ||
          (templateObject_24 = __makeTemplateObject(
            ['\n        transform: translate(0, 50%);\n      '],
            ['\n        transform: translate(0, 50%);\n      ']
          ))
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_25 ||
          (templateObject_25 = __makeTemplateObject(
            ['\n        transform: translate(-8px, 50%);\n      '],
            ['\n        transform: translate(-8px, 50%);\n      ']
          ))
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_26 ||
          (templateObject_26 = __makeTemplateObject(
            ['\n        transform: translate(-12px, 50%);\n      '],
            ['\n        transform: translate(-12px, 50%);\n      ']
          ))
      );

    default:
      return styled_components_1.css(
        templateObject_27 || (templateObject_27 = __makeTemplateObject([''], ['']))
      );
  }
};

exports.generateIconButtonPosition = generateIconButtonPosition;

var generateIconColor = function generateIconColor(_a) {
  var theme = _a.theme,
    disabled = _a.disabled;

  if (disabled) {
    return styled_components_1.css(
      templateObject_28 ||
        (templateObject_28 = __makeTemplateObject(
          ['\n      svg {\n        fill: ', ';\n        cursor: not-allowed;\n      }\n    '],
          ['\n      svg {\n        fill: ', ';\n        cursor: not-allowed;\n      }\n    ']
        )),
      theme.transparent[25]
    );
  }

  return styled_components_1.css(
    templateObject_29 ||
      (templateObject_29 = __makeTemplateObject(
        ['\n    svg {\n      fill: ', ';\n    }\n  '],
        ['\n    svg {\n      fill: ', ';\n    }\n  ']
      )),
    theme.transparent[50]
  );
};

exports.generateIconColor = generateIconColor;
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12,
  templateObject_13,
  templateObject_14,
  templateObject_15,
  templateObject_16,
  templateObject_17,
  templateObject_18,
  templateObject_19,
  templateObject_20,
  templateObject_21,
  templateObject_22,
  templateObject_23,
  templateObject_24,
  templateObject_25,
  templateObject_26,
  templateObject_27,
  templateObject_28,
  templateObject_29;
