// rtk-ui-kit/components/Overlay/Overlay.js
'use strict';
var PropTypes = require('prop-types');

var __assign =
  (void 0 && (void 0).__assign) ||
  function () {
    __assign =
      Object.assign ||
      function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
          s = arguments[i];

          for (var p in s) {
            if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
          }
        }

        return t;
      };

    return __assign.apply(this, arguments);
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var __rest =
  (void 0 && (void 0).__rest) ||
  function (s, e) {
    var t = {};

    for (var p in s) {
      if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    }

    if (s != null && typeof Object.getOwnPropertySymbols === 'function')
      for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
          t[p[i]] = s[p[i]];
      }
    return t;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importStar(require('react'));

var react_dom_1 = require('react-dom');

var function_1 = require('../../utils/function');

var constants_1 = require('./constants');

var styles_1 = require('./styles');

var overlayRoot = typeof window !== 'undefined' ? document.createElement('div') : null;
var Overlay = react_1.forwardRef(function Overlay(props, ref) {
  var _a = props.show,
    show = _a === void 0 ? false : _a,
    _b = props.view,
    view = _b === void 0 ? constants_1.DEFAULT_VIEW : _b,
    _c = props.useInPortal,
    useInPortal = _c === void 0 ? true : _c,
    _d = props.onClick,
    onClick = _d === void 0 ? function_1.noop : _d,
    restProps = __rest(props, ['show', 'view', 'useInPortal', 'onClick']);

  react_1.useEffect(
    function () {
      if (overlayRoot && show && useInPortal) {
        document.body.appendChild(overlayRoot);
      }

      return function () {
        if (overlayRoot && document.body.contains(overlayRoot)) {
          document.body.removeChild(overlayRoot);
        }
      };
    },
    [show, useInPortal]
  );
  var overlay = show
    ? react_1.default.createElement(
        styles_1.StyledOverlay,
        __assign(
          {
            ref: ref,
            view: view,
            onClick: onClick,
          },
          restProps
        )
      )
    : null;
  return useInPortal && overlayRoot ? react_dom_1.createPortal(overlay, overlayRoot) : overlay;
});

Overlay.propTypes = {
  /**
   * Callback функция, вызываемая при клике
   */
  onClick: PropTypes.func,
  /**
   * Ярлык видимости компонента
   *  @default false
   */
  show: PropTypes.bool,
  /**
   * Задает использование в Portal
   *  @default true
   */
  useInPortal: PropTypes.bool,
  /**
   * Устанавливает тип компонента
   *  @default "standart"
   */
  view: PropTypes.oneOf(['soft', 'standart']),
};

Overlay.displayName = 'Overlay';
exports.default = Overlay;
