// rtk-ui-kit/components/Tooltip/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.DEFAULT_TEXT_ALIGN =
  exports.DEFAULT_PLACEMENT =
  exports.DEFAULT_TRIGGER =
  exports.DEFAULT_IMAGE_POSITION =
  exports.DEFAULT_HINT_POSITION =
  exports.MAX_WIDTH =
  exports.TOOLTIP_OFFSET =
  exports.ARROW_OFFSET_CIRCULAR =
  exports.ARROW_OFFSET_ROUNDED =
  exports.ARROW_OFFSET_GEOMETRIC =
  exports.ARROW_HEIGHT =
  exports.ARROW_WIDTH =
  exports.TEXT_ALIGNS =
  exports.TOOLTIP_PLACEMENTS =
  exports.IMAGE_POSITIONS =
  exports.HINT_POSITIONS =
  exports.TOOLTIP_TRIGGERS =
    void 0;
var TOOLTIP_TRIGGERS;

(function (TOOLTIP_TRIGGERS) {
  TOOLTIP_TRIGGERS['click'] = 'click';
  TOOLTIP_TRIGGERS['hover'] = 'hover';
})((TOOLTIP_TRIGGERS = exports.TOOLTIP_TRIGGERS || (exports.TOOLTIP_TRIGGERS = {})));

var HINT_POSITIONS;

(function (HINT_POSITIONS) {
  HINT_POSITIONS['top'] = 'top';
  HINT_POSITIONS['bottom'] = 'bottom';
})((HINT_POSITIONS = exports.HINT_POSITIONS || (exports.HINT_POSITIONS = {})));

var IMAGE_POSITIONS;

(function (IMAGE_POSITIONS) {
  IMAGE_POSITIONS['top'] = 'top';
  IMAGE_POSITIONS['bottom'] = 'bottom';
})((IMAGE_POSITIONS = exports.IMAGE_POSITIONS || (exports.IMAGE_POSITIONS = {})));

var TOOLTIP_PLACEMENTS;

(function (TOOLTIP_PLACEMENTS) {
  TOOLTIP_PLACEMENTS['auto'] = 'auto';
  TOOLTIP_PLACEMENTS['top'] = 'top';
  TOOLTIP_PLACEMENTS['bottom'] = 'bottom';
  TOOLTIP_PLACEMENTS['right'] = 'right';
  TOOLTIP_PLACEMENTS['left'] = 'left';
  TOOLTIP_PLACEMENTS['topLeft'] = 'topLeft';
  TOOLTIP_PLACEMENTS['topRight'] = 'topRight';
  TOOLTIP_PLACEMENTS['bottomLeft'] = 'bottomLeft';
  TOOLTIP_PLACEMENTS['bottomRight'] = 'bottomRight';
  TOOLTIP_PLACEMENTS['leftTop'] = 'leftTop';
  TOOLTIP_PLACEMENTS['leftBottom'] = 'leftBottom';
  TOOLTIP_PLACEMENTS['rightTop'] = 'rightTop';
  TOOLTIP_PLACEMENTS['rightBottom'] = 'rightBottom';
})((TOOLTIP_PLACEMENTS = exports.TOOLTIP_PLACEMENTS || (exports.TOOLTIP_PLACEMENTS = {})));

var TEXT_ALIGNS;

(function (TEXT_ALIGNS) {
  TEXT_ALIGNS['left'] = 'left';
  TEXT_ALIGNS['center'] = 'center';
})((TEXT_ALIGNS = exports.TEXT_ALIGNS || (exports.TEXT_ALIGNS = {})));

var POSITION_OFFSET = 4;
exports.ARROW_WIDTH = 16;
exports.ARROW_HEIGHT = 8;
exports.ARROW_OFFSET_GEOMETRIC = 12 + exports.ARROW_WIDTH / 2;
exports.ARROW_OFFSET_ROUNDED = 16 + exports.ARROW_WIDTH / 2;
exports.ARROW_OFFSET_CIRCULAR = 20 + exports.ARROW_WIDTH / 2;
exports.TOOLTIP_OFFSET = exports.ARROW_HEIGHT + POSITION_OFFSET + 'px';
exports.MAX_WIDTH = 246;
exports.DEFAULT_HINT_POSITION = HINT_POSITIONS.bottom;
exports.DEFAULT_IMAGE_POSITION = IMAGE_POSITIONS.bottom;
exports.DEFAULT_TRIGGER = TOOLTIP_TRIGGERS.click;
exports.DEFAULT_PLACEMENT = TOOLTIP_PLACEMENTS.auto;
exports.DEFAULT_TEXT_ALIGN = TEXT_ALIGNS.left;
