// rtk-ui-kit/components/Tooltip/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledTooltip =
  exports.StyledTooltipNoImage =
  exports.StyledTooltipImage =
  exports.StyledTooltipCloseButton =
  exports.StyledTooltipHint =
  exports.StyledTooltipText =
  exports.StyledTooltipHeader =
  exports.StyledTooltipTrigger =
  exports.StyledTooltipRoot =
    void 0;

var styled_components_1 = __importStar(require('styled-components'));

var styledGenerators_1 = require('./styledGenerators');

var constants_1 = require('./constants');

var CloseButton_1 = __importDefault(require('../Button/CloseButton/CloseButton'));

var themeFonts_1 = require('../../constants/themeFonts');

exports.StyledTooltipRoot = styled_components_1.default.div(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  display: inline-block;\n  vertical-align: middle;\n  position: relative;\n  max-width: fit-content;\n',
      ],
      [
        '\n  display: inline-block;\n  vertical-align: middle;\n  position: relative;\n  max-width: fit-content;\n',
      ]
    ))
);
exports.StyledTooltipTrigger = styled_components_1.default.div(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      ['\n  display: flex;\n  ', ';\n'],
      ['\n  display: flex;\n  ', ';\n']
    )),
  styledGenerators_1.generateDisabledChildren
);
var TextOverflowEllipsis = styled_components_1.css(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      ['\n  overflow: hidden;\n  text-overflow: ellipsis;\n'],
      ['\n  overflow: hidden;\n  text-overflow: ellipsis;\n']
    ))
);
exports.StyledTooltipHeader = styled_components_1.default.div(
  templateObject_4 ||
    (templateObject_4 = __makeTemplateObject(
      ['\n  color: ', ';\n\n  ', '\n  ', '\n'],
      ['\n  color: ', ';\n\n  ', '\n  ', '\n']
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[100];
  },
  function (_a) {
    var theme = _a.theme;
    return theme.fonts.static.bodyM;
  },
  TextOverflowEllipsis
);
exports.StyledTooltipText = styled_components_1.default.div(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      ['\n  color: ', ';\n\n  ', '\n  ', '\n  ', ';\n'],
      ['\n  color: ', ';\n\n  ', '\n  ', '\n  ', ';\n']
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[75];
  },
  function (_a) {
    var theme = _a.theme;
    return theme.fonts.static.caption;
  },
  TextOverflowEllipsis,
  styledGenerators_1.generateTextAlign
);
exports.StyledTooltipHint = styled_components_1.default.div(
  templateObject_6 ||
    (templateObject_6 = __makeTemplateObject(
      [
        '\n  font-size: 10px;\n  line-height: 12px;\n  letter-spacing: 0.4px;\n  color: ',
        ';\n  text-align: center;\n\n  ',
        '\n  ',
        ';\n',
      ],
      [
        '\n  font-size: 10px;\n  line-height: 12px;\n  letter-spacing: 0.4px;\n  color: ',
        ';\n  text-align: center;\n\n  ',
        '\n  ',
        ';\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[50];
  },
  TextOverflowEllipsis,
  themeFonts_1.baseFontSettings
);
exports.StyledTooltipCloseButton = styled_components_1.default(CloseButton_1.default)(
  templateObject_7 ||
    (templateObject_7 = __makeTemplateObject(
      ['\n  position: absolute;\n  top: 16px;\n  right: 16px;\n'],
      ['\n  position: absolute;\n  top: 16px;\n  right: 16px;\n']
    ))
);
var StyledTooltipImageCss = styled_components_1.css(
  templateObject_8 ||
    (templateObject_8 = __makeTemplateObject(
      ['\n  justify-self: center;\n  width: 72px;\n  height: 72px;\n  border-radius: 40px;\n'],
      ['\n  justify-self: center;\n  width: 72px;\n  height: 72px;\n  border-radius: 40px;\n']
    ))
);
exports.StyledTooltipImage = styled_components_1.default.img(
  templateObject_9 || (templateObject_9 = __makeTemplateObject(['\n  ', '\n'], ['\n  ', '\n'])),
  StyledTooltipImageCss
);
exports.StyledTooltipNoImage = styled_components_1.default.div(
  templateObject_10 ||
    (templateObject_10 = __makeTemplateObject(
      ['\n  ', ' {\n    background-color: ', ';\n  }\n'],
      ['\n  ', ' {\n    background-color: ', ';\n  }\n']
    )),
  StyledTooltipImageCss,
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[5];
  }
);
exports.StyledTooltip = styled_components_1.default.div(
  templateObject_11 ||
    (templateObject_11 = __makeTemplateObject(
      [
        '\n  position: absolute;\n  z-index: 1051;\n  display: grid;\n  gap: 12px;\n  padding: 16px;\n  background: ',
        ';\n  box-shadow: ',
        ';\n  width: max-content;\n  max-width: ',
        'px;\n  visibility: hidden;\n  ',
        ';\n  ',
        ';\n  ',
        ';\n  ',
        ";\n\n  &::after {\n    content: '';\n    width: 0;\n    height: 0;\n    position: absolute;\n    border: ",
        'px solid transparent;\n    ',
        ';\n  }\n\n  ',
        ' {\n    + ',
        ', + ',
        ', + ',
        ' {\n      padding-right: 36px;\n    }\n\n    + ',
        ', + ',
        ' {\n      margin: 0 36px;\n    }\n  }\n',
      ],
      [
        '\n  position: absolute;\n  z-index: 1051;\n  display: grid;\n  gap: 12px;\n  padding: 16px;\n  background: ',
        ';\n  box-shadow: ',
        ';\n  width: max-content;\n  max-width: ',
        'px;\n  visibility: hidden;\n  ',
        ';\n  ',
        ';\n  ',
        ';\n  ',
        ";\n\n  &::after {\n    content: '';\n    width: 0;\n    height: 0;\n    position: absolute;\n    border: ",
        'px solid transparent;\n    ',
        ';\n  }\n\n  ',
        ' {\n    + ',
        ', + ',
        ', + ',
        ' {\n      padding-right: 36px;\n    }\n\n    + ',
        ', + ',
        ' {\n      margin: 0 36px;\n    }\n  }\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.tooltip.backgroundColor;
  },
  function (_a) {
    var theme = _a.theme;
    return theme.shadows.bottomL;
  },
  constants_1.MAX_WIDTH,
  styledGenerators_1.generateVisibility,
  styledGenerators_1.generateShapeTooltip,
  styledGenerators_1.generateMinSize,
  styledGenerators_1.generateTooltipPosition,
  constants_1.ARROW_WIDTH / 2,
  styledGenerators_1.generateArrowPosition,
  exports.StyledTooltipCloseButton,
  exports.StyledTooltipHeader,
  exports.StyledTooltipText,
  exports.StyledTooltipHint,
  exports.StyledTooltipImage,
  exports.StyledTooltipNoImage
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11;
