// rtk-ui-kit/components/Tooltip/styledGenerators.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.generateDisabledChildren =
  exports.generateTooltipPosition =
  exports.generateArrowPosition =
  exports.generateTooltipArrowBottom =
  exports.generateTooltipArrowTop =
  exports.generateTooltipArrowRight =
  exports.generateTooltipArrowLeft =
  exports.generateTextAlign =
  exports.generateMinSize =
  exports.generateVisibility =
  exports.generateShapeTooltip =
    void 0;

var styled_components_1 = require('styled-components');

var components_1 = require('../../constants/components');

var constants_1 = require('./constants');

var utils_1 = require('./utils');

var generateShapeTooltip = function generateShapeTooltip(_a) {
  var shape = _a.shape;

  switch (shape) {
    case components_1.SHAPES.geometric:
      return styled_components_1.css(
        templateObject_1 ||
          (templateObject_1 = __makeTemplateObject(
            ['\n        border-radius: 0;\n      '],
            ['\n        border-radius: 0;\n      ']
          ))
      );

    case components_1.SHAPES.rounded:
      return styled_components_1.css(
        templateObject_2 ||
          (templateObject_2 = __makeTemplateObject(
            ['\n        border-radius: 8px;\n      '],
            ['\n        border-radius: 8px;\n      ']
          ))
      );

    case components_1.SHAPES.circular:
      return styled_components_1.css(
        templateObject_3 ||
          (templateObject_3 = __makeTemplateObject(
            ['\n        border-radius: 16px;\n      '],
            ['\n        border-radius: 16px;\n      ']
          ))
      );

    default:
      return '';
  }
};

exports.generateShapeTooltip = generateShapeTooltip;

var generateVisibility = function generateVisibility(_a) {
  var isOpen = _a.isOpen;

  if (isOpen) {
    return styled_components_1.css(
      templateObject_4 ||
        (templateObject_4 = __makeTemplateObject(
          ['\n      visibility: visible;\n    '],
          ['\n      visibility: visible;\n    ']
        ))
    );
  }

  return '';
};

exports.generateVisibility = generateVisibility;

var generateMinSize = function generateMinSize(_a) {
  var closeButton = _a.closeButton;

  if (closeButton) {
    return styled_components_1.css(
      templateObject_5 ||
        (templateObject_5 = __makeTemplateObject(
          ['\n      min-height: 56px;\n    '],
          ['\n      min-height: 56px;\n    ']
        ))
    );
  }

  return '';
};

exports.generateMinSize = generateMinSize;

var generateTextAlign = function generateTextAlign(_a) {
  var textAlign = _a.textAlign;

  switch (textAlign) {
    case constants_1.TEXT_ALIGNS.center:
      return styled_components_1.css(
        templateObject_6 ||
          (templateObject_6 = __makeTemplateObject(
            ['\n        text-align: center;\n      '],
            ['\n        text-align: center;\n      ']
          ))
      );

    case constants_1.TEXT_ALIGNS.left:
      return styled_components_1.css(
        templateObject_7 ||
          (templateObject_7 = __makeTemplateObject(
            ['\n        text-align: left;\n      '],
            ['\n        text-align: left;\n      ']
          ))
      );

    default:
      return '';
  }
};

exports.generateTextAlign = generateTextAlign;

var generateTooltipArrowLeft = function generateTooltipArrowLeft(_a) {
  var shape = _a.shape;
  return styled_components_1.css(
    templateObject_8 ||
      (templateObject_8 = __makeTemplateObject(
        ['\n    left: ', 'px;\n  '],
        ['\n    left: ', 'px;\n  ']
      )),
    utils_1.getTooltipArrowOffset(shape)
  );
};

exports.generateTooltipArrowLeft = generateTooltipArrowLeft;

var generateTooltipArrowRight = function generateTooltipArrowRight(_a) {
  var shape = _a.shape;
  return styled_components_1.css(
    templateObject_9 ||
      (templateObject_9 = __makeTemplateObject(
        ['\n    right: ', 'px;\n  '],
        ['\n    right: ', 'px;\n  ']
      )),
    utils_1.getTooltipArrowOffset(shape)
  );
};

exports.generateTooltipArrowRight = generateTooltipArrowRight;

var generateTooltipArrowTop = function generateTooltipArrowTop(_a) {
  var shape = _a.shape;
  return styled_components_1.css(
    templateObject_10 ||
      (templateObject_10 = __makeTemplateObject(
        ['\n    top: ', 'px;\n  '],
        ['\n    top: ', 'px;\n  ']
      )),
    utils_1.getTooltipArrowOffset(shape)
  );
};

exports.generateTooltipArrowTop = generateTooltipArrowTop;

var generateTooltipArrowBottom = function generateTooltipArrowBottom(_a) {
  var shape = _a.shape;
  return styled_components_1.css(
    templateObject_11 ||
      (templateObject_11 = __makeTemplateObject(
        ['\n    bottom: ', 'px;\n  '],
        ['\n    bottom: ', 'px;\n  ']
      )),
    utils_1.getTooltipArrowOffset(shape)
  );
};

exports.generateTooltipArrowBottom = generateTooltipArrowBottom;

var generateArrowPosition = function generateArrowPosition(_a) {
  var placement = _a.placement,
    theme = _a.theme;
  var baseTop = styled_components_1.css(
    templateObject_12 ||
      (templateObject_12 = __makeTemplateObject(
        [
          '\n    left: 50%;\n    margin-left: -',
          'px;\n    bottom: -',
          'px;\n    border-top-color: ',
          ';\n    border-bottom: none;\n  ',
        ],
        [
          '\n    left: 50%;\n    margin-left: -',
          'px;\n    bottom: -',
          'px;\n    border-top-color: ',
          ';\n    border-bottom: none;\n  ',
        ]
      )),
    constants_1.ARROW_WIDTH / 2,
    constants_1.ARROW_HEIGHT,
    theme.tooltip.backgroundColor
  );
  var baseBottom = styled_components_1.css(
    templateObject_13 ||
      (templateObject_13 = __makeTemplateObject(
        [
          '\n    left: 50%;\n    margin-left: -',
          'px;\n    top: -',
          'px;\n    border-bottom-color: ',
          ';\n    border-top: none;\n  ',
        ],
        [
          '\n    left: 50%;\n    margin-left: -',
          'px;\n    top: -',
          'px;\n    border-bottom-color: ',
          ';\n    border-top: none;\n  ',
        ]
      )),
    constants_1.ARROW_WIDTH / 2,
    constants_1.ARROW_HEIGHT,
    theme.tooltip.backgroundColor
  );
  var baseLeft = styled_components_1.css(
    templateObject_14 ||
      (templateObject_14 = __makeTemplateObject(
        [
          '\n    top: 50%;\n    margin-top: -',
          'px;\n    right: -',
          'px;\n    border-left-color: ',
          ';\n    border-right: none;\n  ',
        ],
        [
          '\n    top: 50%;\n    margin-top: -',
          'px;\n    right: -',
          'px;\n    border-left-color: ',
          ';\n    border-right: none;\n  ',
        ]
      )),
    constants_1.ARROW_WIDTH / 2,
    constants_1.ARROW_HEIGHT,
    theme.tooltip.backgroundColor
  );
  var baseRight = styled_components_1.css(
    templateObject_15 ||
      (templateObject_15 = __makeTemplateObject(
        [
          '\n    top: 50%;\n    margin-top: -',
          'px;\n    left: -',
          'px;\n    border-right-color: ',
          ';\n    border-left: none;\n  ',
        ],
        [
          '\n    top: 50%;\n    margin-top: -',
          'px;\n    left: -',
          'px;\n    border-right-color: ',
          ';\n    border-left: none;\n  ',
        ]
      )),
    constants_1.ARROW_WIDTH / 2,
    constants_1.ARROW_HEIGHT,
    theme.tooltip.backgroundColor
  );

  switch (placement) {
    case constants_1.TOOLTIP_PLACEMENTS.top:
      return baseTop;

    case constants_1.TOOLTIP_PLACEMENTS.topLeft:
      return styled_components_1.css(
        templateObject_16 ||
          (templateObject_16 = __makeTemplateObject(
            ['\n        ', '\n        ', '\n      '],
            ['\n        ', '\n        ', '\n      ']
          )),
        baseTop,
        exports.generateTooltipArrowLeft
      );

    case constants_1.TOOLTIP_PLACEMENTS.topRight:
      return styled_components_1.css(
        templateObject_17 ||
          (templateObject_17 = __makeTemplateObject(
            [
              '\n        ',
              '\n        margin-right: -',
              'px;\n        left: auto;\n        ',
              ';\n      ',
            ],
            [
              '\n        ',
              '\n        margin-right: -',
              'px;\n        left: auto;\n        ',
              ';\n      ',
            ]
          )),
        baseTop,
        constants_1.ARROW_WIDTH / 2,
        exports.generateTooltipArrowRight
      );

    case constants_1.TOOLTIP_PLACEMENTS.bottom:
      return baseBottom;

    case constants_1.TOOLTIP_PLACEMENTS.bottomLeft:
      return styled_components_1.css(
        templateObject_18 ||
          (templateObject_18 = __makeTemplateObject(
            ['\n        ', '\n        ', '\n      '],
            ['\n        ', '\n        ', '\n      ']
          )),
        baseBottom,
        exports.generateTooltipArrowLeft
      );

    case constants_1.TOOLTIP_PLACEMENTS.bottomRight:
      return styled_components_1.css(
        templateObject_19 ||
          (templateObject_19 = __makeTemplateObject(
            [
              '\n        ',
              '\n        margin-right: -',
              'px;\n        left: auto;\n        ',
              ';\n      ',
            ],
            [
              '\n        ',
              '\n        margin-right: -',
              'px;\n        left: auto;\n        ',
              ';\n      ',
            ]
          )),
        baseBottom,
        constants_1.ARROW_WIDTH / 2,
        exports.generateTooltipArrowRight
      );

    case constants_1.TOOLTIP_PLACEMENTS.left:
      return baseLeft;

    case constants_1.TOOLTIP_PLACEMENTS.leftTop:
      return styled_components_1.css(
        templateObject_20 ||
          (templateObject_20 = __makeTemplateObject(
            ['\n        ', '\n        ', ';\n      '],
            ['\n        ', '\n        ', ';\n      ']
          )),
        baseLeft,
        exports.generateTooltipArrowTop
      );

    case constants_1.TOOLTIP_PLACEMENTS.leftBottom:
      return styled_components_1.css(
        templateObject_21 ||
          (templateObject_21 = __makeTemplateObject(
            [
              '\n        ',
              '\n        margin-bottom: -',
              'px;\n        top: auto;\n        ',
              ';\n      ',
            ],
            [
              '\n        ',
              '\n        margin-bottom: -',
              'px;\n        top: auto;\n        ',
              ';\n      ',
            ]
          )),
        baseLeft,
        constants_1.ARROW_WIDTH / 2,
        exports.generateTooltipArrowBottom
      );

    case constants_1.TOOLTIP_PLACEMENTS.right:
      return baseRight;

    case constants_1.TOOLTIP_PLACEMENTS.rightTop:
      return styled_components_1.css(
        templateObject_22 ||
          (templateObject_22 = __makeTemplateObject(
            ['\n        ', '\n        ', ';\n      '],
            ['\n        ', '\n        ', ';\n      ']
          )),
        baseRight,
        exports.generateTooltipArrowTop
      );

    case constants_1.TOOLTIP_PLACEMENTS.rightBottom:
      return styled_components_1.css(
        templateObject_23 ||
          (templateObject_23 = __makeTemplateObject(
            [
              '\n        ',
              '\n        margin-bottom: -',
              'px;\n        top: auto;\n        ',
              ';\n      ',
            ],
            [
              '\n        ',
              '\n        margin-bottom: -',
              'px;\n        top: auto;\n        ',
              ';\n      ',
            ]
          )),
        baseRight,
        constants_1.ARROW_WIDTH / 2,
        exports.generateTooltipArrowBottom
      );

    default:
      return '';
  }
};

exports.generateArrowPosition = generateArrowPosition;

var generateTooltipPosition = function generateTooltipPosition(_a) {
  var placement = _a.placement,
    shape = _a.shape,
    currentTargetRect = _a.currentTargetRect;
  var currentTargetWidthCenter = currentTargetRect.width / 2;
  var currentTargetHeightCenter = currentTargetRect.height / 2;

  switch (placement) {
    case constants_1.TOOLTIP_PLACEMENTS.top:
      return styled_components_1.css(
        templateObject_24 ||
          (templateObject_24 = __makeTemplateObject(
            [
              '\n        bottom: calc(100% + ',
              ');\n        left: 50%;\n        transform: translateX(-50%);\n      ',
            ],
            [
              '\n        bottom: calc(100% + ',
              ');\n        left: 50%;\n        transform: translateX(-50%);\n      ',
            ]
          )),
        constants_1.TOOLTIP_OFFSET
      );

    case constants_1.TOOLTIP_PLACEMENTS.topLeft:
      return styled_components_1.css(
        templateObject_25 ||
          (templateObject_25 = __makeTemplateObject(
            ['\n        bottom: calc(100% + ', ');\n        left: ', ';\n      '],
            ['\n        bottom: calc(100% + ', ');\n        left: ', ';\n      ']
          )),
        constants_1.TOOLTIP_OFFSET,
        utils_1.getTooltipArrowOffset(shape) > currentTargetWidthCenter
          ? -(utils_1.getTooltipArrowOffset(shape) - currentTargetWidthCenter) + 'px'
          : '0'
      );

    case constants_1.TOOLTIP_PLACEMENTS.topRight:
      return styled_components_1.css(
        templateObject_26 ||
          (templateObject_26 = __makeTemplateObject(
            ['\n        bottom: calc(100% + ', ');\n        right: ', ';\n      '],
            ['\n        bottom: calc(100% + ', ');\n        right: ', ';\n      ']
          )),
        constants_1.TOOLTIP_OFFSET,
        utils_1.getTooltipArrowOffset(shape) > currentTargetWidthCenter
          ? -(utils_1.getTooltipArrowOffset(shape) - currentTargetWidthCenter) + 'px'
          : '0'
      );

    case constants_1.TOOLTIP_PLACEMENTS.bottom:
      return styled_components_1.css(
        templateObject_27 ||
          (templateObject_27 = __makeTemplateObject(
            [
              '\n        top: calc(100% + ',
              ');\n        left: 50%;\n        transform: translateX(-50%);\n      ',
            ],
            [
              '\n        top: calc(100% + ',
              ');\n        left: 50%;\n        transform: translateX(-50%);\n      ',
            ]
          )),
        constants_1.TOOLTIP_OFFSET
      );

    case constants_1.TOOLTIP_PLACEMENTS.bottomLeft:
      return styled_components_1.css(
        templateObject_28 ||
          (templateObject_28 = __makeTemplateObject(
            ['\n        top: calc(100% + ', ');\n        left: ', ';\n      '],
            ['\n        top: calc(100% + ', ');\n        left: ', ';\n      ']
          )),
        constants_1.TOOLTIP_OFFSET,
        utils_1.getTooltipArrowOffset(shape) > currentTargetWidthCenter
          ? -(utils_1.getTooltipArrowOffset(shape) - currentTargetWidthCenter) + 'px'
          : '0'
      );

    case constants_1.TOOLTIP_PLACEMENTS.bottomRight:
      return styled_components_1.css(
        templateObject_29 ||
          (templateObject_29 = __makeTemplateObject(
            ['\n        top: calc(100% + ', ');\n        right: ', ';\n      '],
            ['\n        top: calc(100% + ', ');\n        right: ', ';\n      ']
          )),
        constants_1.TOOLTIP_OFFSET,
        utils_1.getTooltipArrowOffset(shape) > currentTargetWidthCenter
          ? -(utils_1.getTooltipArrowOffset(shape) - currentTargetWidthCenter) + 'px'
          : '0'
      );

    case constants_1.TOOLTIP_PLACEMENTS.left:
      return styled_components_1.css(
        templateObject_30 ||
          (templateObject_30 = __makeTemplateObject(
            [
              '\n        right: calc(100% + ',
              ');\n        transform: translateY(calc(-50% - ',
              '));\n      ',
            ],
            [
              '\n        right: calc(100% + ',
              ');\n        transform: translateY(calc(-50% - ',
              '));\n      ',
            ]
          )),
        constants_1.TOOLTIP_OFFSET,
        constants_1.TOOLTIP_OFFSET
      );

    case constants_1.TOOLTIP_PLACEMENTS.leftTop:
      return styled_components_1.css(
        templateObject_31 ||
          (templateObject_31 = __makeTemplateObject(
            ['\n        right: calc(100% + ', ');\n        top: ', ';\n      '],
            ['\n        right: calc(100% + ', ');\n        top: ', ';\n      ']
          )),
        constants_1.TOOLTIP_OFFSET,
        utils_1.getTooltipArrowOffset(shape) > currentTargetHeightCenter
          ? -(utils_1.getTooltipArrowOffset(shape) - currentTargetHeightCenter) + 'px'
          : '0'
      );

    case constants_1.TOOLTIP_PLACEMENTS.leftBottom:
      return styled_components_1.css(
        templateObject_32 ||
          (templateObject_32 = __makeTemplateObject(
            ['\n        right: calc(100% + ', ');\n        bottom: ', ';\n      '],
            ['\n        right: calc(100% + ', ');\n        bottom: ', ';\n      ']
          )),
        constants_1.TOOLTIP_OFFSET,
        utils_1.getTooltipArrowOffset(shape) > currentTargetHeightCenter
          ? -(utils_1.getTooltipArrowOffset(shape) - currentTargetHeightCenter) + 'px'
          : '0'
      );

    case constants_1.TOOLTIP_PLACEMENTS.right:
      return styled_components_1.css(
        templateObject_33 ||
          (templateObject_33 = __makeTemplateObject(
            [
              '\n        left: calc(100% + ',
              ');\n        transform: translateY(calc(-50% - ',
              '));\n      ',
            ],
            [
              '\n        left: calc(100% + ',
              ');\n        transform: translateY(calc(-50% - ',
              '));\n      ',
            ]
          )),
        constants_1.TOOLTIP_OFFSET,
        constants_1.TOOLTIP_OFFSET
      );

    case constants_1.TOOLTIP_PLACEMENTS.rightTop:
      return styled_components_1.css(
        templateObject_34 ||
          (templateObject_34 = __makeTemplateObject(
            ['\n        left: calc(100% + ', ');\n        top: ', ';\n      '],
            ['\n        left: calc(100% + ', ');\n        top: ', ';\n      ']
          )),
        constants_1.TOOLTIP_OFFSET,
        utils_1.getTooltipArrowOffset(shape) > currentTargetHeightCenter
          ? -(utils_1.getTooltipArrowOffset(shape) - currentTargetHeightCenter) + 'px'
          : '0'
      );

    case constants_1.TOOLTIP_PLACEMENTS.rightBottom:
      return styled_components_1.css(
        templateObject_35 ||
          (templateObject_35 = __makeTemplateObject(
            ['\n        left: calc(100% + ', ');\n        bottom: ', ';\n      '],
            ['\n        left: calc(100% + ', ');\n        bottom: ', ';\n      ']
          )),
        constants_1.TOOLTIP_OFFSET,
        utils_1.getTooltipArrowOffset(shape) > currentTargetHeightCenter
          ? -(utils_1.getTooltipArrowOffset(shape) - currentTargetHeightCenter) + 'px'
          : '0'
      );

    default:
      return '';
  }
};

exports.generateTooltipPosition = generateTooltipPosition;

var generateDisabledChildren = function generateDisabledChildren(_a) {
  var hasDisabledChildren = _a.hasDisabledChildren;

  if (hasDisabledChildren) {
    return styled_components_1.css(
      templateObject_36 ||
        (templateObject_36 = __makeTemplateObject(
          [
            '\n      cursor: not-allowed;\n      button:disabled {\n        pointer-events: none;\n      }\n    ',
          ],
          [
            '\n      cursor: not-allowed;\n      button:disabled {\n        pointer-events: none;\n      }\n    ',
          ]
        ))
    );
  }

  return styled_components_1.css(
    templateObject_37 ||
      (templateObject_37 = __makeTemplateObject(
        ['\n    cursor: pointer;\n  '],
        ['\n    cursor: pointer;\n  ']
      ))
  );
};

exports.generateDisabledChildren = generateDisabledChildren;
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8,
  templateObject_9,
  templateObject_10,
  templateObject_11,
  templateObject_12,
  templateObject_13,
  templateObject_14,
  templateObject_15,
  templateObject_16,
  templateObject_17,
  templateObject_18,
  templateObject_19,
  templateObject_20,
  templateObject_21,
  templateObject_22,
  templateObject_23,
  templateObject_24,
  templateObject_25,
  templateObject_26,
  templateObject_27,
  templateObject_28,
  templateObject_29,
  templateObject_30,
  templateObject_31,
  templateObject_32,
  templateObject_33,
  templateObject_34,
  templateObject_35,
  templateObject_36,
  templateObject_37;
