// rtk-ui-kit/components/Header/Logo.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importDefault(require('react'));

var styled_components_1 = __importDefault(require('styled-components'));

var components_1 = require('../../constants/components');

var StyledLogo = styled_components_1.default.svg(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      [
        '\n  > g.logo-text > path {\n    fill: ',
        ';\n  }\n  @media screen and (min-width: ',
        'px) {\n    > .logo-text {\n      display: none;\n    }\n  }\n  @media screen and (min-width: ',
        'px) {\n    > .logo-text {\n      display: block;\n    }\n  }\n',
      ],
      [
        '\n  > g.logo-text > path {\n    fill: ',
        ';\n  }\n  @media screen and (min-width: ',
        'px) {\n    > .logo-text {\n      display: none;\n    }\n  }\n  @media screen and (min-width: ',
        'px) {\n    > .logo-text {\n      display: block;\n    }\n  }\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.transparent[100];
  },
  components_1.BREAKPOINTS.mobile_small,
  components_1.BREAKPOINTS.mobile_large
);

var Logo = function Logo() {
  return react_1.default.createElement(
    StyledLogo,
    {
      width: '166',
      height: '44',
      viewBox: '0 0 166 44',
      fill: 'none',
    },
    react_1.default.createElement(
      'mask',
      {
        id: 'mask0',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '0',
        y: '33',
        width: '19',
        height: '11',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M18.9223 43.5961H3.00392C2.53602 43.5961 2.08094 43.443 1.70808 43.1601C0.764293 42.4445 0.57937 41.0993 1.29506 40.1557L8.25201 33.2749C8.25218 33.2749 11.809 36.7153 18.9223 43.5961Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        mask: 'url(#mask0)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M-3.44136 28.9741H23.2228V47.8964H-3.44136V28.9741Z',
        fill: '#FF4F12',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        id: 'mask1',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '0',
        y: '0',
        width: '26',
        height: '44',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M0 40.1556V13.7642C0 12.21 0.60104 10.716 1.67746 9.59439L10.3212 0.59082L25.8031 16.0727L1.93609 39.9543C1.53468 40.3543 1.30899 40.8979 1.30899 41.4647C1.30899 42.6421 2.26328 43.596 3.44041 43.596C1.54036 43.596 0 42.0556 0 40.1556Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        mask: 'url(#mask1)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M-4.30052 -3.70972H30.1036V47.8965H-4.30052V-3.70972Z',
        fill: '#7800FF',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask2',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '124',
        y: '27',
        width: '13',
        height: '13',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M136.131 27.4038L130.434 33.1502L136.281 39.797H132.483L127.961 34.6751V39.797H124.812V27.4038H127.961V31.9013L132.408 27.4038H136.131Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask2)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M120.512 23.103H140.581V44.0973H120.512V23.103Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask3',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '28',
        y: '22',
        width: '15',
        height: '18',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M28.3834 39.7968V22.3066H36.3287C40.2765 22.3066 42.875 24.3305 42.875 28.0031C42.875 31.6766 40.2765 33.6996 36.3287 33.6996H31.6813V39.7968H28.3834ZM36.2289 30.752C38.5274 30.752 39.5769 29.6519 39.5769 28.0031C39.5769 26.3543 38.5274 25.2551 36.2289 25.2551H31.6813V30.752H36.2289Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask3)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M24.0829 18.0063H47.1756V44.0976H24.0829V18.0063Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask4',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '43',
        y: '27',
        width: '15',
        height: '14',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M50.3194 37.1978C52.1684 37.1978 53.8924 35.8741 53.8924 33.6C53.8924 31.3267 52.1684 30.0021 50.3194 30.0021C48.4707 30.0021 46.7466 31.3267 46.7466 33.6C46.7466 35.8741 48.4707 37.1978 50.3194 37.1978ZM50.3194 27.0537C53.7176 27.0537 57.0906 29.253 57.0906 33.6C57.0906 37.9478 53.7176 40.1462 50.3194 40.1462C46.9215 40.1462 43.5483 37.9478 43.5483 33.6C43.5483 29.253 46.9215 27.0537 50.3194 27.0537Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask4)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M39.2477 22.7534H61.3911V44.447H39.2477V22.7534Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask5',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '58',
        y: '27',
        width: '13',
        height: '14',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M65.1281 37.1978C66.6771 37.1978 67.4768 36.4486 68.2263 35.3993L70.8248 36.8236C69.5753 38.9472 67.8016 40.1462 65.1531 40.1462C61.4803 40.1462 58.6317 37.5478 58.6317 33.6C58.6317 29.6521 61.6301 27.0537 65.1531 27.0537C67.8514 27.0537 69.8005 28.2278 70.6998 30.1518L68.0014 31.7017C67.2269 30.5268 66.2525 30.0021 65.1281 30.0021C63.2791 30.0021 61.83 31.4015 61.83 33.6C61.83 35.8242 63.2292 37.1978 65.1281 37.1978Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask5)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M54.3312 22.7534H75.1253V44.447H54.3312V22.7534Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask6',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '96',
        y: '27',
        width: '13',
        height: '13',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M98.6482 34.4248L99.1729 27.4038H108.567V39.7721H105.419V30.2275H101.871L101.572 34.5495C101.322 38.2222 100.322 39.9467 97.6488 39.9467C96.9243 39.9467 96.5329 39.8271 96.5329 39.8271L96.5172 36.6198C96.821 36.6998 97.0297 36.7488 97.4735 36.7488C98.1231 36.7488 98.4985 36.3239 98.6482 34.4248Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask6)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M92.2167 23.103H112.868V44.247H92.2167V23.103Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask7',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '83',
        y: '27',
        width: '13',
        height: '14',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M86.2705 31.9511H92.8418C92.5421 30.4021 91.3678 29.4525 89.6186 29.4525C87.9696 29.4525 86.7204 30.5517 86.2705 31.9511ZM89.7187 27.0537C93.2666 27.0537 95.765 29.628 95.765 33.2508V34.35H86.2207C86.4205 36.2491 87.7699 37.4481 89.6186 37.4481C91.2926 37.4481 91.9922 36.7239 92.7667 35.7494L95.2653 36.9982C94.0161 39.1218 92.2922 40.1462 89.5687 40.1462C86.1208 40.1462 83.1974 37.5229 83.1974 33.6C83.1974 29.6779 86.1208 27.0537 89.7187 27.0537Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask7)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M78.8969 22.7534H100.066V44.447H78.8969V22.7534Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask8',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '110',
        y: '27',
        width: '13',
        height: '14',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M113.403 31.9511H119.974C119.674 30.4021 118.5 29.4525 116.751 29.4525C115.102 29.4525 113.853 30.5517 113.403 31.9511ZM116.851 27.0537C120.399 27.0537 122.898 29.628 122.898 33.2508V34.35H113.353C113.553 36.2491 114.902 37.4481 116.751 37.4481C118.425 37.4481 119.125 36.7239 119.899 35.7494L122.398 36.9982C121.148 39.1218 119.424 40.1462 116.701 40.1462C113.253 40.1462 110.33 37.5229 110.33 33.6C110.33 29.6779 113.253 27.0537 116.851 27.0537Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask8)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M106.029 22.7534H127.198V44.447H106.029V22.7534Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask9',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '71',
        y: '27',
        width: '12',
        height: '13',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M71.8041 30.1518V27.4038H82.348V30.1518H78.6499V39.797H75.502V30.1518H71.8041Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask9)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M67.5036 23.103H86.6486V44.0973H67.5036V23.103Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask10',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '135',
        y: '27',
        width: '15',
        height: '14',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M142.531 37.1986C144.38 37.1986 146.104 35.8741 146.104 33.6C146.104 31.3267 144.38 30.0021 142.531 30.0021C140.682 30.0021 138.958 31.3267 138.958 33.6C138.958 35.8741 140.682 37.1986 142.531 37.1986ZM142.531 27.0537C145.929 27.0537 149.302 29.253 149.302 33.6C149.302 37.9478 145.929 40.1462 142.531 40.1462C139.133 40.1462 135.76 37.9478 135.76 33.6C135.76 29.253 139.133 27.0537 142.531 27.0537Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask10)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M131.459 22.7534H153.603V44.447H131.459V22.7534Z',
        fill: '#101828',
      })
    ),
    react_1.default.createElement(
      'mask',
      {
        className: 'logo-text',
        id: 'mask11',
        'mask-type': 'alpha',
        maskUnits: 'userSpaceOnUse',
        x: '150',
        y: '27',
        width: '16',
        height: '13',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M161.829 27.4014H165.944V39.7722H162.951V31.941L159.634 39.7722H157.19L153.873 31.941V39.7722H150.88V27.4014H154.995L158.412 36.0807L161.829 27.4014Z',
        fill: 'white',
      })
    ),
    react_1.default.createElement(
      'g',
      {
        className: 'logo-text',
        mask: 'url(#mask11)',
      },
      react_1.default.createElement('path', {
        fillRule: 'evenodd',
        clipRule: 'evenodd',
        d: 'M146.58 23.1011H170.244V44.073H146.58V23.1011Z',
        fill: '#101828',
      })
    )
  );
};

exports.default = Logo;
var templateObject_1;
