// rtk-ui-kit/components/wrappers/ThemeProvider/ThemeProvider.js
'use strict';
var PropTypes = require('prop-types');

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importDefault(require('react'));

var styled_components_1 = require('styled-components');

var generateTheme_1 = __importDefault(require('../../../utils/generateTheme'));

var ThemeContext_1 = require('./ThemeContext');

var ThemeProvider = function ThemeProvider(_a) {
  var children = _a.children,
    themeName = _a.themeName,
    themeConfig = _a.themeConfig;
  var generatedTheme = generateTheme_1.default(themeName, themeConfig);
  return react_1.default.createElement(
    styled_components_1.ThemeProvider,
    {
      theme: generatedTheme,
    },
    react_1.default.createElement(
      ThemeContext_1.ThemeContext.Provider,
      {
        value: {
          themeName: themeName,
          themeConfig: generatedTheme,
        },
      },

      children
    )
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node,
  themeConfig: PropTypes.shape({
    baseColors: PropTypes.shape({
      primary1: PropTypes.string,
      primary2: PropTypes.string,
      secondary1: PropTypes.string,
      secondary2: PropTypes.string,
    }),
    greyScale: PropTypes.shape({
      10: PropTypes.string,
      20: PropTypes.string,
      40: PropTypes.string,
      5: PropTypes.string,
      60: PropTypes.string,
      80: PropTypes.string,
      90: PropTypes.string,
      95: PropTypes.string,
      black: PropTypes.string,
      white: PropTypes.string,
    }),
    shape: PropTypes.oneOf(['circular', 'geometric', 'rounded']),
  }),
  themeName: PropTypes.oneOf(['dark', 'light']).isRequired,
};

exports.default = ThemeProvider;
