// rtk-ui-kit/components/Notifications/ToastNotification/ToastNotification.js
'use strict';
var PropTypes = require('prop-types');

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importDefault(require('react'));

var components_1 = require('../../../constants/components');

var constants_1 = require('./constants');

var constants_2 = require('../constants');

var useThemeContext_1 = require('../../../hooks/useThemeContext');

var NotificationButton_1 = __importDefault(
  require('../components/NotificationButton/NotificationButton')
);

var CustomToastNotification_1 = __importDefault(
  require('../CustomToastNotification/CustomToastNotification')
);

var ToastNotification = function ToastNotification(props) {
  var themeConfig = useThemeContext_1.useThemeContext().themeConfig;
  var _a = props.type,
    type = _a === void 0 ? components_1.SYSTEM_COLORS.info : _a,
    _b = props.shape,
    shape = _b === void 0 ? themeConfig.shape : _b,
    title = props.title,
    subtitle = props.subtitle,
    _c = props.id,
    id = _c === void 0 ? constants_2.DEFAULT_ID : _c,
    timeout = props.timeout,
    _d = props.actionButtons,
    actionButtons = _d === void 0 ? [] : _d,
    onOpen = props.onOpen,
    onClose = props.onClose,
    hideCloseButton = props.hideCloseButton,
    position = props.position,
    className = props.className;
  var Icon = constants_1.TOAST_ICON_COMPONENT[type];
  var notificationIcon =
    Icon &&
    react_1.default.createElement(Icon, {
      size: constants_2.ICON_SIZE,
    });

  var notificationButtons = actionButtons.map(function (button) {
    return react_1.default.createElement(NotificationButton_1.default, {
      id: id,
      text: button.text,
      action: function action() {
        return button.action(id);
      },
    });
  });
  return react_1.default.createElement(CustomToastNotification_1.default, {
    shape: shape,
    icon: notificationIcon,
    iconBgColor: type,
    title: title,
    subtitle: subtitle,
    id: id,
    timeout: timeout,
    actionButtons: function actionButtons() {
      return notificationButtons;
    },
    onOpen: onOpen,
    onClose: onClose,
    hideCloseButton: hideCloseButton,
    position: position,
    className: className,
  });
};

ToastNotification.propTypes = {
  /**
   * Задает кнопки уведомления
   * @param {string} text - Текст кнопки
   * @param {(id: string) => void} action - Callback функция, вызываемая при клике
   */
  actionButtons: PropTypes.arrayOf(
    PropTypes.shape({
      action: PropTypes.func.isRequired,
      text: PropTypes.string.isRequired,
    })
  ),
  /**
   * Задает дополнительные классы для компонента
   */
  className: PropTypes.string,
  /**
   * Прячет кнопку "Закрыть"
   *  @default false
   */
  hideCloseButton: PropTypes.bool,
  /**
   * Задает id для компонента
   */
  id: PropTypes.string,
  /**
   * Callback функция, вызываемая при закрытии уведомления
   * @returns {undefined}
   */
  onClose: PropTypes.func,
  /**
   * Callback функция, вызываемая при появлении уведомления
   * @returns {undefined}
   */
  onOpen: PropTypes.func,
  /**
   * Задает расположение уведомления
   */
  position: PropTypes.oneOf([
    'bottomCenter',
    'bottomLeft',
    'bottomRight',
    'topCenter',
    'topLeft',
    'topRight',
  ]),
  /**
   * Задает скругление, значение по умолчанию берется из темы
   */
  shape: PropTypes.oneOf(['circular', 'geometric', 'rounded']),
  /**
   * Задает подзаголовок уведомления
   */
  subtitle: PropTypes.string,
  /**
   * Задает время в милисекундах, через которое уведомление скроется
   */
  timeout: PropTypes.number,
  /**
   * Задает заголовок уведомления
   */
  title: PropTypes.string,
  /**
   * Задает основной цвет для уведомления
   *  @default "info"
   */
  type: PropTypes.oneOf(['error', 'info', 'success', 'warning']),
};

exports.default = ToastNotification;
