// rtk-ui-kit/components/Stepper/styledGenerators.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var _a, _b;

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.generateStepperActiveBtnBackground =
  exports.generateStepperHoveredBtnBackground =
  exports.generateTextFont =
  exports.generateDisabledBtn =
  exports.generateInlineFlex =
  exports.generateStepperSize =
  exports.generateStepperShape =
  exports.generateColor =
    void 0;

var styled_components_1 = require('styled-components');

var components_1 = require('../../constants/components');

var stepper_1 = require('../../constants/stepper');

var constants_1 = require('./constants');

var STEPPER_SHAPES_LIST =
  ((_a = {}),
  (_a[components_1.SHAPES.geometric] = styled_components_1.css(
    templateObject_1 ||
      (templateObject_1 = __makeTemplateObject(
        ['\n    border-radius: 0;\n  '],
        ['\n    border-radius: 0;\n  ']
      ))
  )),
  (_a[components_1.SHAPES.rounded] = styled_components_1.css(
    templateObject_2 ||
      (templateObject_2 = __makeTemplateObject(
        ['\n    border-radius: 8px;\n  '],
        ['\n    border-radius: 8px;\n  ']
      ))
  )),
  (_a[components_1.SHAPES.circular] = styled_components_1.css(
    templateObject_3 ||
      (templateObject_3 = __makeTemplateObject(
        ['\n    border-radius: 24px;\n  '],
        ['\n    border-radius: 24px;\n  ']
      ))
  )),
  _a);
var STEPPER_TEXT_SIZES_LIST =
  ((_b = {}),
  (_b[stepper_1.STEPPER_SIZES.medium] = function (_a) {
    var theme = _a.theme;
    return theme.fonts.static.accentM;
  }),
  (_b[stepper_1.STEPPER_SIZES.small] = function (_a) {
    var theme = _a.theme;
    return theme.fonts.static.accentS;
  }),
  _b);

var generateColor = function generateColor(_a) {
  var propsColor = _a.color,
    currentTheme = _a.currentTheme,
    _b = _a.theme,
    greyScale = _b.greyScale,
    stepperTransparent = _b.stepper.transparent,
    transparent = _b.transparent,
    disabled = _a.disabled;
  var color = greyScale.black;

  if (propsColor || currentTheme === components_1.THEMES.dark) {
    color = greyScale.white;
  }

  if (disabled) {
    color = propsColor ? stepperTransparent.default : transparent[25];
  }

  return color;
};

exports.generateColor = generateColor;

var generateStepperShape = function generateStepperShape(_a) {
  var shape = _a.shape;
  return STEPPER_SHAPES_LIST[shape];
};

exports.generateStepperShape = generateStepperShape;

var generateStepperSize = function generateStepperSize(_a) {
  var size = _a.size,
    text = _a.text;
  var width = constants_1.STEPPER_WIDTH_LIST[size];
  var height = constants_1.STEPPER_HEIGHT_LIST[size];

  if (text && width && height) {
    return styled_components_1.css(
      templateObject_4 ||
        (templateObject_4 = __makeTemplateObject(
          ['\n      min-width: ', 'px;\n      height: ', 'px;\n    '],
          ['\n      min-width: ', 'px;\n      height: ', 'px;\n    ']
        )),
      width,
      height
    );
  }

  return width && height
    ? styled_components_1.css(
        templateObject_5 ||
          (templateObject_5 = __makeTemplateObject(
            ['\n        width: ', 'px;\n        height: ', 'px;\n      '],
            ['\n        width: ', 'px;\n        height: ', 'px;\n      ']
          )),
        width,
        height
      )
    : '';
};

exports.generateStepperSize = generateStepperSize;

var generateInlineFlex = function generateInlineFlex() {
  return styled_components_1.css(
    templateObject_6 ||
      (templateObject_6 = __makeTemplateObject(
        ['\n  display: inline-flex;\n  align-items: center;\n  justify-content: center;\n'],
        ['\n  display: inline-flex;\n  align-items: center;\n  justify-content: center;\n']
      ))
  );
};

exports.generateInlineFlex = generateInlineFlex;

var generateDisabledBtn = function generateDisabledBtn(_a) {
  var disabled = _a.disabled;

  if (disabled) {
    return '\n      cursor: not-allowed;\n  \n      &:hover {\n        background-color: transparent;\n      }\n      ';
  }

  return '';
};

exports.generateDisabledBtn = generateDisabledBtn;

var generateTextFont = function generateTextFont(_a) {
  var size = _a.size;
  return STEPPER_TEXT_SIZES_LIST[size];
};

exports.generateTextFont = generateTextFont;

var generateStepperHoveredBtnBackground = function generateStepperHoveredBtnBackground(_a) {
  var theme = _a.theme,
    disabled = _a.disabled;
  return disabled ? '' : '\nbackground-color: ' + theme.transparent[5] + ';\n';
};

exports.generateStepperHoveredBtnBackground = generateStepperHoveredBtnBackground;

var generateStepperActiveBtnBackground = function generateStepperActiveBtnBackground(_a) {
  var theme = _a.theme,
    disabled = _a.disabled;
  return disabled ? '' : '\nbackground-color: ' + theme.transparent[10] + ';\n';
};

exports.generateStepperActiveBtnBackground = generateStepperActiveBtnBackground;
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6;
