// rtk-ui-kit/components/Multiselect/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledMultiSelectHint =
  exports.MultiselectTagGroup =
  exports.StyledMultiselectInput =
  exports.StyledMultiselectInputContainer =
  exports.StyledMultiselectContainer =
  exports.StyledIcon =
    void 0;

var styled_components_1 = __importDefault(require('styled-components'));

var styledGenerators_1 = require('./styledGenerators');

var InputMaster_1 = __importDefault(require('../Input/InputMaster'));

var TagGroup_1 = __importDefault(require('../Tag/TagGroup/TagGroup'));

var Typography_1 = __importDefault(require('../Typography/Typography'));

var navigation_1 = require('../Icons/navigation');

exports.StyledIcon = styled_components_1.default(navigation_1.ChevronDown)(
  templateObject_1 ||
    (templateObject_1 = __makeTemplateObject(
      ['\n  transform: ', ';\n  transition: transform 300ms;\n'],
      ['\n  transform: ', ';\n  transition: transform 300ms;\n']
    )),
  function (_a) {
    var isOpen = _a.isOpen;
    return isOpen ? 'rotate(180deg)' : 'rotate(0deg)';
  }
);
exports.StyledMultiselectContainer = styled_components_1.default.div(
  templateObject_2 ||
    (templateObject_2 = __makeTemplateObject(
      [
        '\n  position: relative;\n  min-width: 280px;\n\n  display: flex;\n  flex-direction: column;\n',
      ],
      [
        '\n  position: relative;\n  min-width: 280px;\n\n  display: flex;\n  flex-direction: column;\n',
      ]
    ))
);
exports.StyledMultiselectInputContainer = styled_components_1.default.div(
  templateObject_3 ||
    (templateObject_3 = __makeTemplateObject(
      [
        '\n  display: flex;\n\n  &:hover {\n    cursor: ',
        ';\n\n    ',
        ' {\n      fill: ',
        ';\n    }\n  }\n',
      ],
      [
        '\n  display: flex;\n\n  &:hover {\n    cursor: ',
        ';\n\n    ',
        ' {\n      fill: ',
        ';\n    }\n  }\n',
      ]
    )),
  function (_a) {
    var disabled = _a.disabled;
    return disabled ? 'not-allowed' : 'pointer';
  },
  exports.StyledIcon,
  function (_a) {
    var theme = _a.theme,
      disabled = _a.disabled;
    return disabled ? theme.transparent[25] : theme.transparent[100];
  }
);
exports.StyledMultiselectInput = styled_components_1.default(InputMaster_1.default)(
  templateObject_4 ||
    (templateObject_4 = __makeTemplateObject(
      ['\n  width: 100%;\n\n  > input {\n    height: 100%;\n  }\n'],
      ['\n  width: 100%;\n\n  > input {\n    height: 100%;\n  }\n']
    ))
);
exports.MultiselectTagGroup = styled_components_1.default(TagGroup_1.default)(
  templateObject_5 ||
    (templateObject_5 = __makeTemplateObject(
      [
        '\n  width: 100%;\n  padding: 26px 16px 6px 0;\n  ',
        ';\n  min-height: 48px;\n  margin-left: -100%;\n  z-index: 1;\n  box-sizing: border-box;\n\n  div {\n    overflow: hidden;\n    margin-left: 0;\n    margin-right: 4px;\n    mask-image: ',
        ';\n    ',
        ';\n\n    div {\n      margin-top: 4px;\n      padding-left: 4px;\n      padding-right: 0;\n      overflow: unset;\n      mask-image: unset;\n    }\n  }\n\n  ',
        ';\n',
      ],
      [
        '\n  width: 100%;\n  padding: 26px 16px 6px 0;\n  ',
        ';\n  min-height: 48px;\n  margin-left: -100%;\n  z-index: 1;\n  box-sizing: border-box;\n\n  div {\n    overflow: hidden;\n    margin-left: 0;\n    margin-right: 4px;\n    mask-image: ',
        ';\n    ',
        ';\n\n    div {\n      margin-top: 4px;\n      padding-left: 4px;\n      padding-right: 0;\n      overflow: unset;\n      mask-image: unset;\n    }\n  }\n\n  ',
        ';\n',
      ]
    )),
  styledGenerators_1.generateTagsGroupRightPadding,
  function (_a) {
    var theme = _a.theme;
    return 'linear-gradient(to right, ' + theme.greyScale.black + ' 90%, transparent)';
  },
  styledGenerators_1.generateTagsWrap,
  styledGenerators_1.generateShapeTagsPadding
);
exports.StyledMultiSelectHint = styled_components_1.default(Typography_1.default)(
  templateObject_6 ||
    (templateObject_6 = __makeTemplateObject(
      [
        '\n  margin-top: 8px;\n  max-width: inherit;\n  word-break: break-word;\n\n  ',
        ';\n  ',
        ';\n',
      ],
      [
        '\n  margin-top: 8px;\n  max-width: inherit;\n  word-break: break-word;\n\n  ',
        ';\n  ',
        ';\n',
      ]
    )),
  styledGenerators_1.generateHintColor,
  styledGenerators_1.generateGap
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6;
