// rtk-ui-kit/components/PickerDate/CalendarMonths/CalendarMonths.js
'use strict';
var PropTypes = require('prop-types');

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importDefault(require('react'));

var constants_1 = require('../constants');

var useGetBaseColorFromProps_1 = require('../../../hooks/useGetBaseColorFromProps');

var useThemeContext_1 = require('../../../hooks/useThemeContext');

var styles_1 = require('../styles');

var styles_2 = require('./styles');

var CalendarMonths = function CalendarMonths(props) {
  var themeConfig = useThemeContext_1.useThemeContext().themeConfig;
  var _a = props.shape,
    shape = _a === void 0 ? themeConfig.shape : _a,
    _b = props.color,
    color = _b === void 0 ? constants_1.DEFAULT_COLOR : _b,
    date = props.date,
    _c = props.months,
    months = _c === void 0 ? constants_1.MONTHS : _c,
    onSelect = props.onSelect;
  var baseColorFromProps = useGetBaseColorFromProps_1.useGetBaseColorFromProps(color);

  var handleClick = function handleClick(index) {
    return function (e) {
      e.nativeEvent.stopImmediatePropagation();
      onSelect(index);
    };
  };

  return react_1.default.createElement(
    styles_2.StyledCalendarMonths,
    null,
    months.map(function (month, index) {
      return react_1.default.createElement(
        styles_1.StyledCalendarSelectebleItem,
        {
          key: month,
          shape: shape,
          active: index === date.getMonth(),
          onClick: handleClick(index),
          baseColor: baseColorFromProps,
        },
        month
      );
    })
  );
};

CalendarMonths.propTypes = {
  color: PropTypes.oneOf(['primary1', 'primary2', 'secondary1', 'secondary2']),
  date: PropTypes.instanceOf(Date).isRequired,
  months: PropTypes.arrayOf(PropTypes.string),
  onSelect: PropTypes.func.isRequired,
  shape: PropTypes.oneOf(['circular', 'geometric', 'rounded']),
};

CalendarMonths.displayName = 'CalendarMonths';
exports.default = CalendarMonths;
