// rtk-ui-kit/components/Sticker/Sticker.js
'use strict';
var PropTypes = require('prop-types');

var __importDefault =
  (void 0 && (void 0).__importDefault) ||
  function (mod) {
    return mod && mod.__esModule
      ? mod
      : {
          default: mod,
        };
  };

Object.defineProperty(exports, '__esModule', {
  value: true,
});

var react_1 = __importDefault(require('react'));

var useThemeContext_1 = require('../../hooks/useThemeContext');

var styles_1 = require('./styles');

var Sticker = function Sticker(props) {
  var themeConfig = useThemeContext_1.useThemeContext().themeConfig;
  var className = props.className,
    style = props.style,
    disabled = props.disabled,
    _a = props.shape,
    shape = _a === void 0 ? themeConfig.shape : _a,
    children = props.children,
    type = props.type;
  return react_1.default.createElement(
    styles_1.StyledStickerContainer,
    {
      className: className,
      style: style,
      disabled: disabled,
      shape: shape,
      type: type,
    },
    react_1.default.createElement(styles_1.StyledStickerText, null, children)
  );
};

Sticker.propTypes = {
  children: PropTypes.node,
  /**
   * Устанавливает дополнительные классы для элемента
   */
  className: PropTypes.string,
  /**
   * Устанавливает атрибут disabled
   */
  disabled: PropTypes.bool,
  /**
   * Задает скругление, значение по умолчанию берется из темы
   */
  shape: PropTypes.oneOf(['circular', 'geometric', 'rounded']),
  /**
   * Задает дополнительные стили для компонента
   */
  style: PropTypes.object,
  /**
   * Устанавливает тип отображаемого компонента
   */
  type: PropTypes.oneOf([
    'error',
    'info',
    'primary1',
    'primary2',
    'secondary1',
    'secondary2',
    'success',
    'warning',
  ]).isRequired,
};

Sticker.displayName = 'Sticker';
exports.default = Sticker;
