// rtk-ui-kit/components/Sticker/styles.js
'use strict';

var __makeTemplateObject =
  (void 0 && (void 0).__makeTemplateObject) ||
  function (cooked, raw) {
    if (Object.defineProperty) {
      Object.defineProperty(cooked, 'raw', {
        value: raw,
      });
    } else {
      cooked.raw = raw;
    }

    return cooked;
  };

var __createBinding =
  (void 0 && (void 0).__createBinding) ||
  (Object.create
    ? function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        Object.defineProperty(o, k2, {
          enumerable: true,
          get: function get() {
            return m[k];
          },
        });
      }
    : function (o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
      });

var __setModuleDefault =
  (void 0 && (void 0).__setModuleDefault) ||
  (Object.create
    ? function (o, v) {
        Object.defineProperty(o, 'default', {
          enumerable: true,
          value: v,
        });
      }
    : function (o, v) {
        o['default'] = v;
      });

var __importStar =
  (void 0 && (void 0).__importStar) ||
  function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod) {
        if (k !== 'default' && Object.prototype.hasOwnProperty.call(mod, k))
          __createBinding(result, mod, k);
      }

    __setModuleDefault(result, mod);

    return result;
  };

var _a;

Object.defineProperty(exports, '__esModule', {
  value: true,
});

exports.StyledStickerText = exports.StyledStickerContainer = void 0;

var styled_components_1 = __importStar(require('styled-components'));

var themeUtils_1 = require('../../utils/themeUtils');

var components_1 = require('../../constants/components');

var themeFonts_1 = require('../../constants/themeFonts');

var STICKER_SHAPES =
  ((_a = {}),
  (_a[components_1.SHAPES.geometric] = styled_components_1.css(
    templateObject_1 ||
      (templateObject_1 = __makeTemplateObject(
        ['\n    border-radius: 0;\n  '],
        ['\n    border-radius: 0;\n  ']
      ))
  )),
  (_a[components_1.SHAPES.rounded] = styled_components_1.css(
    templateObject_2 ||
      (templateObject_2 = __makeTemplateObject(
        ['\n    border-radius: 4px;\n  '],
        ['\n    border-radius: 4px;\n  ']
      ))
  )),
  (_a[components_1.SHAPES.circular] = styled_components_1.css(
    templateObject_3 ||
      (templateObject_3 = __makeTemplateObject(
        ['\n    border-radius: 16px;\n  '],
        ['\n    border-radius: 16px;\n  ']
      ))
  )),
  _a);

var generateShape = function generateShape(_a) {
  var shape = _a.shape;
  return STICKER_SHAPES[shape];
};

var generateBackground = function generateBackground(_a) {
  var theme = _a.theme,
    type = _a.type;
  var systemColor = themeUtils_1.getThemedSystemColors(type, theme);
  var baseColor = themeUtils_1.getThemedBaseColors(type, theme);
  var color = baseColor || systemColor;
  return color
    ? styled_components_1.css(
        templateObject_4 ||
          (templateObject_4 = __makeTemplateObject(
            ['\n        background-color: ', ';\n      '],
            ['\n        background-color: ', ';\n      ']
          )),
        color
      )
    : '';
};

var generateDisabled = function generateDisabled(_a) {
  var disabled = _a.disabled,
    theme = _a.theme;
  return disabled
    ? styled_components_1.css(
        templateObject_5 ||
          (templateObject_5 = __makeTemplateObject(
            ['\n        background-color: ', ';\n        color: ', ';\n      '],
            ['\n        background-color: ', ';\n        color: ', ';\n      ']
          )),
        theme.transparent[10],
        theme.transparent[25]
      )
    : '';
}; // Исключение, шрифт не из темы

var generateFont = function generateFont() {
  return styled_components_1.css(
    templateObject_6 ||
      (templateObject_6 = __makeTemplateObject(
        ['\n  ', '\n  font-weight: 500;\n  font-size: 10px;\n  letter-spacing: 0.4px;\n'],
        ['\n  ', '\n  font-weight: 500;\n  font-size: 10px;\n  letter-spacing: 0.4px;\n']
      )),
    themeFonts_1.baseFontSettings
  );
};

exports.StyledStickerContainer = styled_components_1.default.div(
  templateObject_7 ||
    (templateObject_7 = __makeTemplateObject(
      [
        '\n  box-sizing: border-box;\n  display: inline-flex;\n  align-items: center;\n  padding: 2px 4px;\n  max-width: 100%;\n  height: 16px;\n  color: ',
        ';\n  ',
        '\n  ',
        '\n  ',
        '\n  ',
        '\n',
      ],
      [
        '\n  box-sizing: border-box;\n  display: inline-flex;\n  align-items: center;\n  padding: 2px 4px;\n  max-width: 100%;\n  height: 16px;\n  color: ',
        ';\n  ',
        '\n  ',
        '\n  ',
        '\n  ',
        '\n',
      ]
    )),
  function (_a) {
    var theme = _a.theme;
    return theme.greyScale.white;
  },
  generateFont,
  generateShape,
  generateBackground,
  generateDisabled
);
exports.StyledStickerText = styled_components_1.default.span(
  templateObject_8 ||
    (templateObject_8 = __makeTemplateObject(
      ['\n  overflow: hidden;\n  text-overflow: ellipsis;\n  white-space: nowrap;\n'],
      ['\n  overflow: hidden;\n  text-overflow: ellipsis;\n  white-space: nowrap;\n']
    ))
);
var templateObject_1,
  templateObject_2,
  templateObject_3,
  templateObject_4,
  templateObject_5,
  templateObject_6,
  templateObject_7,
  templateObject_8;
